<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-02-06" />

<title>Explore the APA qtls from RNAseq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Explore the APA qtls from RNAseq</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2018-02-06</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-02-12</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4e6265c</p>
<!-- Add your analysis here -->
<pre class="r"><code>library(&quot;dplyr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<p>The goal of this analysis is to do some exploration on the APA qtls that Yang called on the LCL RNA-seq data.</p>
<p>Relevant files are in: /project2/gilad/yangili/LCLs</p>
<p>run_lm_APA.txt: APA QTLs. cols are: ensembl geneid, snpID, distance to target, pvalue, beta</p>
<p>qqnorm_APA_final.txt.gz : what went into the qtl mapping</p>
<p>DaPars_APA_geuvadis.txt : the “raw” APA usage</p>
<p>ensembl2refseq.txt : ensembl ids to the refseq ids</p>
<p>First step: Use deeptools to look at APA enrichment in some of the individuals This is the script for 18585. /project2/gilad/briana/apa_sites/rnaseq_LCL/heat_map18505.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=deeptools_lcl
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3

source activate net-seq

bamCoverage -b /project2/gilad/yangili/LCLs/bams/RNAseqGeuvadis_STAR_18505.final.bam -o LCL_18505.bw

computeMatrix reference-point -S LCL_18505.bw -R /project2/gilad/briana/apa_sites/clusters.bed -b 1000 -a 1000 -out LCL_18505.gz

plotHeatmap --sortRegions --xAxisLabel &#39;APA&#39;  descend -m LCL_18505.gz  -out LCL_18505.apa.png</code></pre>
<p><strong>Need to index the bam files before I can run this</strong></p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=sort_bam
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END



module load Anaconda3  

source activate net-seq 


sample=$1
describer=$(echo ${sample} | sed -e &quot;s/\.bam$//&quot;)
#describer=$(echo ${sample} | sed &#39;s/.bam//&#39;)  

# Sort BAM file  
samtools sort ${describer}.bam -o ${describer}.sort.bam  


   </code></pre>
<pre class="bash"><code>
#!/bin/bash


#SBATCH --job-name=index_bams
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3  

source activate net-seq 


sample=$1
describer=$(echo ${sample} | sed &#39;s/.bam//&#39;)  


# index the bam file  
samtools index ${describer}.bam  
</code></pre>
<p>Run these 2 scripts for:</p>
<ul>
<li><p>RNAseqGeuvadis_STAR_18505.final.bam</p></li>
<li><p>RNAseqGeuvadis_STAR_18486.final.bam</p></li>
</ul>
<p>Now I can run the deeptools script:</p>
<div class="figure">
<img src="../data/LCL_18505.apa.png" alt="APA enrichment in 18505 RNA seq" />
<p class="caption">APA enrichment in 18505 RNA seq</p>
</div>
<p>How to fix the TSS label!</p>
<p><strong>–refPointLabel APA</strong></p>
<div id="take-a-look-at-the-qtl-files" class="section level3">
<h3>Take a look at the QTL files:</h3>
<pre class="r"><code>#qtl file
APA_qtls=read.table(&quot;../data/run_lm_APA.txt&quot;, stringsAsFactors = FALSE)
names(APA_qtls)=c(&quot;gene&quot;, &quot;snp&quot;, &quot;distance&quot;, &quot;pval&quot;, &quot;beta&quot;)

#APA usage
APA_usage=read.table(&quot;../data/DaPars_APA_geuvadis.txt&quot;, stringsAsFactors = FALSE)
names(APA_usage)=c(&quot;#Chr&quot;,  &quot;start&quot;,    &quot;end&quot;,  &quot;ID&quot;,   &quot;NA18486&quot;,  &quot;NA18487&quot;,&quot;NA18488&quot;,    &quot;NA18489&quot;,  &quot;NA18498&quot;,  &quot;NA18499&quot;,  &quot;NA18502&quot;,  &quot;NA18505&quot;,  &quot;NA18508&quot;,  &quot;NA18510&quot;,  &quot;NA18511&quot;,  &quot;NA18517&quot;,  &quot;NA18519&quot;,  &quot;NA18520&quot;,  &quot;NA18858&quot;,  &quot;NA18861&quot;,  &quot;NA18867&quot;,  &quot;NA18868&quot;,  &quot;NA18870&quot;,  &quot;NA18873&quot;,  &quot;NA18907&quot;,  &quot;NA18909&quot;,  &quot;NA18910&quot;,  &quot;NA18912&quot;,  &quot;NA18916&quot;,  &quot;NA18917&quot;,  &quot;NA18923&quot;,  &quot;NA18933&quot;,  &quot;NA18934&quot;,  &#39;NA19092&#39;,  &quot;NA19093&quot;,  &quot;NA19095&quot;,  &quot;NA19096&quot;,  &quot;NA19098&quot;,  &quot;NA19099&quot;,  &quot;NA19102&quot;,  &quot;NA19107&quot;,  &quot;NA19108&quot;,  &#39;NA19113&#39;,  &#39;NA19114&#39;,  &#39;NA19116&#39;,  &#39;NA19117&#39;,  &#39;NA19118&#39;,  &#39;NA19119&#39;,  &#39;NA19121&#39;,  &#39;NA19129&#39;,  &#39;NA19130&#39;, &#39;NA19131&#39;,   &#39;NA19137&#39;,  &#39;NA19138&#39;,  &#39;NA19141&#39;,  &#39;NA19143&#39;,  &#39;NA19144&#39;,  &#39;NA19146&#39;,  &#39;NA19147&#39;,  &#39;NA19149&#39;,  &#39;NA19150&#39;, &#39;NA19152&#39;,   &#39;NA19153&#39;,  &#39;NA19159&#39;,  &#39;NA19160&#39;,  &#39;NA19171&#39;, &#39;NA19172&#39;,   &#39;NA19175&#39;,  &#39;NA19184&#39;,  &#39;NA19185&#39;,  &#39;NA19189&#39;,  &#39;NA19190&#39;,  &#39;NA19197&#39;,  &#39;NA19198&#39;,  &#39;NA19200&#39;,  &#39;NA19201&#39;,  &#39;NA19204&#39;,  &#39;NA19206&#39;,  &#39;NA19207&#39;,  &#39;NA19209&#39;,  &#39;NA19210&#39;,  &#39;NA19213&#39;,  &#39;NA19214&#39;,  &#39;NA19222&#39;,  &#39;NA19223&#39;,  &#39;NA19225&#39;,  &#39;NA19235&#39;,  &#39;NA19247&#39;,  &#39;NA19248&#39;)

#gene IDs
ensenbl2refseq=read.delim(&quot;../data/ensembl2refseq.txt&quot;, stringsAsFactors = FALSE)</code></pre>
<p>To look at top QTLs I will sort by pval:</p>
<pre class="r"><code>APA_qtls=arrange(APA_qtls, APA_qtls$pval )

top_QTL=APA_qtls[1,]
top_QTL</code></pre>
<pre><code>             gene       snp distance        pval  beta
1 ENSG00000258289 rs7144811   -19710 2.10997e-37 1.322</code></pre>
<p>Find this gene in the ensenbl2refseq file:</p>
<pre class="r"><code>ref_gene_top=ensenbl2refseq[which(grepl(top_QTL[1], ensenbl2refseq[,1])),]
ref_gene_top</code></pre>
<pre><code>      Ensembl.Gene.ID Ensembl.Transcript.ID
44380 ENSG00000258289       ENST00000552002
44396 ENSG00000258289       ENST00000549115
44407 ENSG00000258289       ENST00000548752
44422 ENSG00000258289       ENST00000551947
44427 ENSG00000258289       ENST00000556089
44438 ENSG00000258289       ENST00000551093
44443 ENSG00000258289       ENST00000547625
44454 ENSG00000258289       ENST00000359118
      RefSeq.mRNA..e.g..NM_001195597.
44380                       NM_145165
44396                    NM_001204063
44407                    NM_001204064
44422                                
44427                                
44438                                
44443                                
44454                                
      RefSeq.mRNA.predicted..e.g..XM_001125684.
44380                                          
44396                                          
44407                                          
44422                                          
44427                                          
44438                                          
44443                                          
44454                                          </code></pre>
<p>Find the APA usage line for this gene.</p>
<pre class="r"><code>top_apa_usage=APA_usage[which(grepl(&quot;NM_001204064&quot;,APA_usage[,4])),]
top_apa_usage</code></pre>
<pre><code>      #Chr    start      end                          ID NA18486 NA18487
9370 chr14 65398856 65402084 NM_001204064|CHURC1|chr14|+    0.17    0.63
     NA18488 NA18489 NA18498 NA18499 NA18502 NA18505 NA18508 NA18510
9370    0.95    0.43    0.16    0.63    0.12    0.75    0.71    0.37
     NA18511 NA18517 NA18519 NA18520 NA18858 NA18861 NA18867 NA18868
9370    0.09    0.03     0.1       1     0.2     0.8    0.18    0.69
     NA18870 NA18873 NA18907 NA18909 NA18910 NA18912 NA18916 NA18917
9370    0.28    0.55    0.39    0.06    0.45    0.72    0.42    0.47
     NA18923 NA18933 NA18934 NA19092 NA19093 NA19095 NA19096 NA19098
9370    0.41    0.28    0.71    0.21    0.07    0.48    0.14     0.7
     NA19099 NA19102 NA19107 NA19108 NA19113 NA19114 NA19116 NA19117
9370    0.28     0.5    0.42    0.35    0.17     0.4    0.68     0.7
     NA19118 NA19119 NA19121 NA19129 NA19130 NA19131 NA19137 NA19138
9370    0.67    0.37    0.14    0.09    0.09    0.13    0.93    0.43
     NA19141 NA19143 NA19144 NA19146 NA19147 NA19149 NA19150 NA19152
9370    0.31    0.47    0.13    0.18    0.34     0.5    0.41    0.28
     NA19153 NA19159 NA19160 NA19171 NA19172 NA19175 NA19184 NA19185
9370    0.93    0.34    0.57    0.14    0.13    0.12    0.83    0.26
     NA19189 NA19190 NA19197 NA19198 NA19200 NA19201 NA19204 NA19206
9370    0.46    0.56    0.36    0.17    0.43    0.22    0.13    0.08
     NA19207 NA19209 NA19210 NA19213 NA19214 NA19222 NA19223 NA19225
9370    0.17    0.69    0.29    0.14    0.13    0.11    0.41    0.67
     NA19235 NA19247 NA19248
9370       1    0.72    0.11</code></pre>
<pre class="r"><code>hist(as.numeric(top_apa_usage[5:89]))</code></pre>
<p><img src="figure/APA_qtl_RNAseq.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(as.numeric(top_apa_usage[5:89]))</code></pre>
<pre><code>[1] 85</code></pre>
<p>The snp I am looking at is rs7144811.</p>
<p>I now need to assign the genotypes to the individuals and APA.</p>
<pre class="r"><code>rs7144811_geno= read.table(&quot;../data/genotypes.rs7144811.txt&quot;)
names_geno=read.delim(&quot;../data/names_geno.txt&quot;, header=FALSE)

names(rs7144811_geno)=unname(unlist(names_geno[1,]))</code></pre>
<pre class="r"><code>individuals=names(APA_usage[,5:ncol(APA_usage)])
individuals </code></pre>
<pre><code> [1] &quot;NA18486&quot; &quot;NA18487&quot; &quot;NA18488&quot; &quot;NA18489&quot; &quot;NA18498&quot; &quot;NA18499&quot; &quot;NA18502&quot;
 [8] &quot;NA18505&quot; &quot;NA18508&quot; &quot;NA18510&quot; &quot;NA18511&quot; &quot;NA18517&quot; &quot;NA18519&quot; &quot;NA18520&quot;
[15] &quot;NA18858&quot; &quot;NA18861&quot; &quot;NA18867&quot; &quot;NA18868&quot; &quot;NA18870&quot; &quot;NA18873&quot; &quot;NA18907&quot;
[22] &quot;NA18909&quot; &quot;NA18910&quot; &quot;NA18912&quot; &quot;NA18916&quot; &quot;NA18917&quot; &quot;NA18923&quot; &quot;NA18933&quot;
[29] &quot;NA18934&quot; &quot;NA19092&quot; &quot;NA19093&quot; &quot;NA19095&quot; &quot;NA19096&quot; &quot;NA19098&quot; &quot;NA19099&quot;
[36] &quot;NA19102&quot; &quot;NA19107&quot; &quot;NA19108&quot; &quot;NA19113&quot; &quot;NA19114&quot; &quot;NA19116&quot; &quot;NA19117&quot;
[43] &quot;NA19118&quot; &quot;NA19119&quot; &quot;NA19121&quot; &quot;NA19129&quot; &quot;NA19130&quot; &quot;NA19131&quot; &quot;NA19137&quot;
[50] &quot;NA19138&quot; &quot;NA19141&quot; &quot;NA19143&quot; &quot;NA19144&quot; &quot;NA19146&quot; &quot;NA19147&quot; &quot;NA19149&quot;
[57] &quot;NA19150&quot; &quot;NA19152&quot; &quot;NA19153&quot; &quot;NA19159&quot; &quot;NA19160&quot; &quot;NA19171&quot; &quot;NA19172&quot;
[64] &quot;NA19175&quot; &quot;NA19184&quot; &quot;NA19185&quot; &quot;NA19189&quot; &quot;NA19190&quot; &quot;NA19197&quot; &quot;NA19198&quot;
[71] &quot;NA19200&quot; &quot;NA19201&quot; &quot;NA19204&quot; &quot;NA19206&quot; &quot;NA19207&quot; &quot;NA19209&quot; &quot;NA19210&quot;
[78] &quot;NA19213&quot; &quot;NA19214&quot; &quot;NA19222&quot; &quot;NA19223&quot; &quot;NA19225&quot; &quot;NA19235&quot; &quot;NA19247&quot;
[85] &quot;NA19248&quot;</code></pre>
<pre class="r"><code>length(individuals)</code></pre>
<pre><code>[1] 85</code></pre>
<p>I need to subset the genotype files to only include these individuals:</p>
<pre class="r"><code>geno_apa_rs7144811=rs7144811_geno[,which((names(rs7144811_geno) %in% individuals)==TRUE)] 
dim(geno_apa_rs7144811)</code></pre>
<pre><code>[1]  1 85</code></pre>
<pre class="r"><code>top_usage_nometa=top_apa_usage[,5:ncol(top_apa_usage)]


#i need to order these the same before I can merge them into one df
names(top_usage_nometa)</code></pre>
<pre><code> [1] &quot;NA18486&quot; &quot;NA18487&quot; &quot;NA18488&quot; &quot;NA18489&quot; &quot;NA18498&quot; &quot;NA18499&quot; &quot;NA18502&quot;
 [8] &quot;NA18505&quot; &quot;NA18508&quot; &quot;NA18510&quot; &quot;NA18511&quot; &quot;NA18517&quot; &quot;NA18519&quot; &quot;NA18520&quot;
[15] &quot;NA18858&quot; &quot;NA18861&quot; &quot;NA18867&quot; &quot;NA18868&quot; &quot;NA18870&quot; &quot;NA18873&quot; &quot;NA18907&quot;
[22] &quot;NA18909&quot; &quot;NA18910&quot; &quot;NA18912&quot; &quot;NA18916&quot; &quot;NA18917&quot; &quot;NA18923&quot; &quot;NA18933&quot;
[29] &quot;NA18934&quot; &quot;NA19092&quot; &quot;NA19093&quot; &quot;NA19095&quot; &quot;NA19096&quot; &quot;NA19098&quot; &quot;NA19099&quot;
[36] &quot;NA19102&quot; &quot;NA19107&quot; &quot;NA19108&quot; &quot;NA19113&quot; &quot;NA19114&quot; &quot;NA19116&quot; &quot;NA19117&quot;
[43] &quot;NA19118&quot; &quot;NA19119&quot; &quot;NA19121&quot; &quot;NA19129&quot; &quot;NA19130&quot; &quot;NA19131&quot; &quot;NA19137&quot;
[50] &quot;NA19138&quot; &quot;NA19141&quot; &quot;NA19143&quot; &quot;NA19144&quot; &quot;NA19146&quot; &quot;NA19147&quot; &quot;NA19149&quot;
[57] &quot;NA19150&quot; &quot;NA19152&quot; &quot;NA19153&quot; &quot;NA19159&quot; &quot;NA19160&quot; &quot;NA19171&quot; &quot;NA19172&quot;
[64] &quot;NA19175&quot; &quot;NA19184&quot; &quot;NA19185&quot; &quot;NA19189&quot; &quot;NA19190&quot; &quot;NA19197&quot; &quot;NA19198&quot;
[71] &quot;NA19200&quot; &quot;NA19201&quot; &quot;NA19204&quot; &quot;NA19206&quot; &quot;NA19207&quot; &quot;NA19209&quot; &quot;NA19210&quot;
[78] &quot;NA19213&quot; &quot;NA19214&quot; &quot;NA19222&quot; &quot;NA19223&quot; &quot;NA19225&quot; &quot;NA19235&quot; &quot;NA19247&quot;
[85] &quot;NA19248&quot;</code></pre>
<pre class="r"><code>names(geno_apa_rs7144811)</code></pre>
<pre><code> [1] &quot;NA18486&quot; &quot;NA18489&quot; &quot;NA18498&quot; &quot;NA18499&quot; &quot;NA18502&quot; &quot;NA18505&quot; &quot;NA18508&quot;
 [8] &quot;NA18510&quot; &quot;NA18511&quot; &quot;NA18517&quot; &quot;NA18519&quot; &quot;NA18520&quot; &quot;NA18858&quot; &quot;NA18861&quot;
[15] &quot;NA18870&quot; &quot;NA18907&quot; &quot;NA18909&quot; &quot;NA18912&quot; &quot;NA18916&quot; &quot;NA19093&quot; &quot;NA19098&quot;
[22] &quot;NA19099&quot; &quot;NA19102&quot; &quot;NA19108&quot; &quot;NA19114&quot; &quot;NA19116&quot; &quot;NA19119&quot; &quot;NA19129&quot;
[29] &quot;NA19131&quot; &quot;NA19137&quot; &quot;NA19138&quot; &quot;NA19141&quot; &quot;NA19143&quot; &quot;NA19144&quot; &quot;NA19147&quot;
[36] &quot;NA19152&quot; &quot;NA19153&quot; &quot;NA19159&quot; &quot;NA19160&quot; &quot;NA19171&quot; &quot;NA19172&quot; &quot;NA19190&quot;
[43] &quot;NA19200&quot; &quot;NA19201&quot; &quot;NA19204&quot; &quot;NA19207&quot; &quot;NA19209&quot; &quot;NA19210&quot; &quot;NA19225&quot;
[50] &quot;NA19175&quot; &quot;NA19095&quot; &quot;NA19096&quot; &quot;NA18867&quot; &quot;NA18868&quot; &quot;NA18923&quot; &quot;NA18488&quot;
[57] &quot;NA19121&quot; &quot;NA19247&quot; &quot;NA19248&quot; &quot;NA18933&quot; &quot;NA18934&quot; &quot;NA19118&quot; &quot;NA19117&quot;
[64] &quot;NA19149&quot; &quot;NA19150&quot; &quot;NA18487&quot; &quot;NA19235&quot; &quot;NA19189&quot; &quot;NA18910&quot; &quot;NA18917&quot;
[71] &quot;NA19113&quot; &quot;NA19214&quot; &quot;NA19213&quot; &quot;NA19146&quot; &quot;NA19107&quot; &quot;NA19185&quot; &quot;NA19184&quot;
[78] &quot;NA19197&quot; &quot;NA19198&quot; &quot;NA18873&quot; &quot;NA19222&quot; &quot;NA19223&quot; &quot;NA19130&quot; &quot;NA19206&quot;
[85] &quot;NA19092&quot;</code></pre>
<pre class="r"><code>geno_apa_rs7144811_sort=geno_apa_rs7144811[,order(names(geno_apa_rs7144811))]
top_usage_nometa_sort=top_usage_nometa[,order(names(top_usage_nometa))]
summary(names(geno_apa_rs7144811_sort)==names(top_usage_nometa_sort))</code></pre>
<pre><code>   Mode    TRUE 
logical      85 </code></pre>
<pre class="r"><code>rs7144811_geno_APA= rbind(top_usage_nometa_sort,geno_apa_rs7144811_sort)
rownames(rs7144811_geno_APA)=c(&quot;APA&quot;, &quot;geno&quot;)</code></pre>
<p>Create a boxplot:</p>
<pre class="r"><code>temp &lt;- rs7144811_geno_APA %&gt;%  tibble::rownames_to_column(., var =&quot;feature&quot;) %&gt;% tidyr::gather(.,Sample,feature)
colnames(temp) &lt;- c(&quot;Feature_name&quot;, &quot;SampleID&quot;, &quot;Value&quot;)
longform &lt;- temp %&gt;% tidyr::spread(.,Feature_name, Value)

#test &lt;- data.frame(apa_score=as.character(rs7144811_geno_APA[1,]), genotype=as.character(rs7144811_geno_APA[2,]), individual=as.vector(colnames(rs7144811_geno_APA)))

rs7144811_plot = ggplot(data = longform) + 
  geom_boxplot(aes(x=geno, group=geno, y=APA), fill=&quot;#4271AE&quot;) +
  labs(title=&quot;APA QTL rs7144811&quot;, x=&quot;Genotype&quot;, y=&quot;APA usage&quot;) 
  



rs7144811_plot</code></pre>
<p><img src="figure/APA_qtl_RNAseq.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /> Think about another way to visualize.</p>
<ol style="list-style-type: decimal">
<li>Look for the PAS in this gene (NM_001204064/ENSG0000025828) it is on the positive strand</li>
</ol>
<ul>
<li>chr14 65398856, 65402084</li>
</ul>
<pre class="r"><code>clusters= read.table(&quot;../data/clusters.bed&quot;)
clusters_chr14=clusters[clusters$V1==&quot;chr14&quot;,]
gene_loc=c(65398856, 65402084)
pas_gene=clusters_chr14[clusters_chr14$V2 &gt; gene_loc[1] &amp; clusters_chr14$V2 &lt; 65402084 &amp; clusters_chr14$V6==&quot;+&quot;,]</code></pre>
<p>I need to create genome coverage -counts. But first I need to sort and index the bam files. I will create a script in bash with a for loop calling /project2/gilad/yangili/LCLs/sort_bams.sh on each file in the directory.</p>
<p>2 files to do this:</p>
<ul>
<li>sort_bam.sh</li>
</ul>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=sort_bam
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END



module load Anaconda3  

source activate net-seq 


sample=$1

describer=$(echo ${sample} | sed -e &quot;s/\.bam$//&quot;)
#describer=$(echo ${sample} | sed -e &#39;.bam//&#39;)
  

echo $sample
echo $describer

# Sort BAM file  
samtools sort ${describer}.bam -o ${describer}.sort.bam</code></pre>
<ul>
<li>wrap_sort_bams.sh</li>
</ul>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=wrap_sort
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3

source activate net-seq 

FILES=bams/*

for i in $FILES ; do
                echo $i
        sbatch /project2/gilad/yangili/LCLs/sort_bams.sh $(echo $i)
        done
</code></pre>
<ul>
<li>wrap_index_bams.sh</li>
</ul>
<pre class="bash"><code>
#!/bin/bash


#SBATCH --job-name=wrap_index
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3

source activate net-seq 

FILES=bams/*.sort.bam

for i in $FILES ; do
                echo $i
        sbatch /project2/gilad/yangili/LCLs/index_bams.sh $(echo $i)
        done
</code></pre>
<p>Script to run genome coverage. /project2/gilad/briana/apa_sites/rnaseq_LCL/count_cov_apa.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=count_cov
#SBATCH --output=count_cov_sbatch.out
#SBATCH --error=count_cov_sbatch.err
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END



module load Anaconda3  

source activate net-seq 


bedtools coverage -counts -sorted -a  /project2/gilad/briana/apa_sites/clusters.sort.bed  -b /project2/gilad/yangili/LCLs/bams/*sort.bam &gt; APA_LCL.coverage.bed
</code></pre>
<p>Waiting to run this until I can re-sort and index RNAseqGeuvadis_STAR_18487.final.bam - Remode this file due to truncated file error.</p>
<p>This code sums the results for all of the bam files. I want a column per samples. I should be able to do this by running it through a wrapper.</p>
<pre class="bash"><code>
#!/bin/bash


#SBATCH --job-name=count_cov
#SBATCH --output=count_cov_sbatch.out
#SBATCH --error=count_cov_sbatch.err
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3  

source activate net-seq 


sample=$1


describer=$(echo ${sample} | sed -e &#39;s/.*\RNAseqGeuvadis_//&#39; | sed -e &quot;s/\.bam$//&quot;)
echo ${desriber}
echo ${sample}

bedtools coverage -counts -sorted -a /project2/gilad/briana/apa_sites/clusters.FAIDXsort.bed -b $1 &gt; /project2/gilad/briana/apa_sites/rnaseq_LCL/cov/${describer}.APA.coverage.bed
</code></pre>
<p>I would pass in one sorted bam file to this script. I will move this script to where the data is to see if this helps with the permision problems. This is now in /project2/gilad/yangili/LCLs</p>
<p>ERROR: chromomsome sort ordering for file bams/RNAseqGeuvadis_STAR_18486.final.sort.bam is inconsistent with other files. Record was: chr1 14466 14541 ERR188214.17353972/1 255 +</p>
<p>I can move it back now. THe permission problem was an extra space. I do need to resort the bed file to match the bam files.</p>
<p>The chromosomes are in order:</p>
<p>10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 1, 20, 21, 22, 2, 3, 4, 5, 6, 7, 8, 9, M, X, Y</p>
<pre class="bash"><code>bedtools sort -faidx /project2/gilad/briana/apa_sites/rnaseq_LCL/names.txt -i /project2/gilad/briana/apa_sites/clusters.sort.bed &gt; /project2/gilad/briana/apa_sites/clusters.FAIDXsort.bed  </code></pre>
<p>The script works now. I need to create a wrap script.</p>
<p>wrap_cov.sh</p>
<pre class="bash"><code>
#!/bin/bash


#SBATCH --job-name=wrap_cov
#SBATCH --time=8:00:00
#SBATCH --partition=broawdl
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3

source activate net-seq 

FILES=bams/*.sort.bam

for i in $FILES ; do
                echo $i
                  sbatch /project2/gilad/briana/apa_sites/rnaseq_LCL/count_cov_apa.sh $(echo $i)
        done
</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2  ggplot2_2.2.1 dplyr_0.7.4  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15     knitr_1.18       bindr_0.1        magrittr_1.5    
 [5] tidyselect_0.2.3 munsell_0.4.3    colorspace_1.3-2 R6_2.2.2        
 [9] rlang_0.1.6      plyr_1.8.4       stringr_1.2.0    tools_3.4.2     
[13] grid_3.4.2       gtable_0.2.0     git2r_0.21.0     htmltools_0.3.6 
[17] lazyeval_0.2.1   yaml_2.1.16      rprojroot_1.3-2  digest_0.6.14   
[21] assertthat_0.2.0 tibble_1.4.2     purrr_0.2.4      tidyr_0.7.2     
[25] glue_1.2.0       evaluate_0.10.1  rmarkdown_1.8.5  stringi_1.1.6   
[29] compiler_3.4.2   pillar_1.1.0     scales_0.5.0     backports_1.1.2 
[33] pkgconfig_2.0.1 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
