<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2016-10-26" />

<title>Set up my Snake and Config files</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Set up my Snake and Config files</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2016-10-26</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-11-01</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 50638e9</p>
<p>The snake and config files are included in the net_seq_pipeline folder. This directory has the human reference genome and the conda environment file.</p>
<p>before each session: <code>module load Anaconda3</code> <code>source activate net-seq</code></p>
<p>To test my snake file: <code>snakemake -np</code></p>
<div id="set-up-config-file" class="section level3">
<h3>Set up config file:</h3>
<pre><code># Snakemake configuration file

# Specify paths to data files

# Paths must end with forward slash

#project directory 
dir_proj: /project2/gilad/briana/net_seq_pipeline/

#directory with aditional scripts:
scripts: /project2/gilad/briana/net_seq_pipeline/scripts/

# Make sure to also update path for log files in cluster.json
dir_log: log/

# Specify Ensembl release for genome sequence and annotation
# http://feb2014.archive.ensembl.org/index.html
ensembl_archive: feb2014.archive.ensembl.org
ensembl_rel: 75
ensembl_genome: GRCh37.75

#Specify chromosomes
#
# For quantifying gene expression
chr_genes: [&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;,
            &quot;10&quot;, &quot;11&quot;, &quot;12&quot;, &quot;13&quot;, &quot;14&quot;, &quot;15&quot;, &quot;16&quot;,
            &quot;17&quot;, &quot;18&quot;, &quot;19&quot;, &quot;20&quot;, &quot;21&quot;, &quot;22&quot;,
            &quot;X&quot;, &quot;Y&quot;, &quot;MT&quot;]
</code></pre>
<p>Make this file executable:</p>
<pre><code>chmod 700 config.yaml</code></pre>
</div>
<div id="set-up-snakefile" class="section level3">
<h3>Set up snakefile:</h3>
<pre><code>#Snakefile
#
#This file will run the net-seq pipeline from fastq files including assembling reference genome
#
#To configure the paths to data files and other settings, edit
#config.yaml
#
#to configure job submission settings for cluster, edit
#cluster.json and submit.snakemake.sh

#to run on RCC midway2 use &#39;bash submit-snakemake.sh&#39;


import glob
import os
from snakemake.utils import R

#Configuration -------------------------------------  


configfile:  &quot;config.yaml&quot;


#Specifu Ensembl release for genome sequence and annotation
ensembl_archive = config[&quot;ensembl_archive&quot;]
ensembl_rel = config[&quot;ensembl_rel&quot;]
ensembl_ftp = &quot;ftp://ftp.ensembl.org/pub/release-&quot; + \
              str(ensembl_rel) + \
              &quot;/fasta/homo_sapiens/dna/&quot;
ensembl_exons = &quot;exons-ensembl-release-&quot; + str(ensembl_rel) + &quot;.saf&quot;
ensembl_genome = config[&quot;ensembl_genome&quot;]

#Paths for data (end with forward slash)
dir_proj=config[&quot;dir_proj&quot;]
dir_data= dir_proj + &quot;data/&quot;
dir_genome= dir_proj + &quot;genome/&quot;
output= dir_proj + &quot;output/&quot;
fastq_dir= dir_data + &quot;fastq/&quot;
fastqc_dir = output + &quot;fastqc/&quot;
chr_genes = config[&quot;chr_genes&quot;]



assert os.path.exists(dir_proj), &quot;Project directory exists&quot;

#Directory to send log files. Needs to be created manually since it
#is not a file created by a Snakemake rule.
dir_log = config[&quot;dir_log&quot;] 
if not os.path.isdir(dir_log):
     os.mkdir(dir_log)

</code></pre>
</div>
<div id="rules-to-add-to-my-snake-file" class="section level2">
<h2>Rules to add to my snake file</h2>
<div id="rules-to-download-and-index-genome" class="section level3">
<h3>Rules to download and index genome</h3>
<p>This is from John’s cardioQTL snake file</p>
<pre><code>rule download_genome:
    output: dir_genome + &quot;Homo_sapiens.&quot; + ensembl_genome + \
            &quot;.dna_sm.chromosome.{chr}.fa.gz&quot;
    params: chr = &quot;{chr}&quot;, build = ensembl_genome
    shell: &quot;wget -O {output} {ensembl_ftp}Homo_sapiens.{params.build}.dna_sm.chromosome.{params.chr}.fa.gz&quot;

rule unzip_chromosome_fasta:
    input: dir_genome + &quot;Homo_sapiens.&quot; + ensembl_genome + \
           &quot;.dna_sm.chromosome.{chr}.fa.gz&quot;
    output: temp(dir_genome + &quot;Homo_sapiens.&quot; + ensembl_genome + \
                 &quot;.dna_sm.chromosome.{chr}.fa&quot;)
    shell: &quot;zcat {input} &gt; {output}&quot;

rule subread_index:
    input: expand(dir_genome + &quot;Homo_sapiens.&quot; + ensembl_genome + \
                  &quot;.dna_sm.chromosome.{chr}.fa&quot;, \
                  chr = chr_genes)
    output: dir_genome + ensembl_genome + &quot;.reads&quot;
    params: prefix = dir_genome + ensembl_genome
    shell: &quot;subread-buildindex -o {params.prefix} {input}&quot;</code></pre>
</div>
<div id="fastqc-rule" class="section level3">
<h3>Fastqc rule</h3>
<p>First add a path to the fastqc dir.</p>
<pre><code>fastqc_dir = output + &quot;fastqc/</code></pre>
<pre><code>rule fastqc:
    input:
        fastq + &quot;{read_num}.fastq&quot;
    output:
        fastqc_dir + &quot;{read_num}_fastqc.html&quot;,
        fastqc_dir + &quot;{read_num}_fastqc.zip&quot;
    params:
        outdir = fastqc_dir
    shell:
        &quot;fastqc -o {params.outdir} {input}&quot;</code></pre>
</div>
<div id="unzip-fastqc" class="section level3">
<h3>Unzip fastqc</h3>
<p>Maybe I dont need to do this right now: This is code to maybe add later:</p>
<pre><code>rule unzip:
    input:
        fastqc_dir + &quot;{read_num}_fastqc.zip&quot;
    output:
        fastqc_dir + &quot;{read_num}_fastqc.zip/fastqc_data.txt&quot;
    params:
        outdir = fastqc_dir
    shell:
        &quot;unzip -d {params.outdir} {input}&quot;</code></pre>
</div>
<div id="extract-umi-data" class="section level3">
<h3>Extract umi data</h3>
<p>This is the next step to figure out. I need to trim the primer and extract the barcode from the linker.</p>
<p>Barcode DNA linker /5rApp/(N)6CTGTAGGCACCATCAAT/3ddC</p>
<p>Use UMI tools to extract first 6 for each read</p>
<p>info from help:</p>
<ul>
<li>need whitelist</li>
</ul>
<pre><code>rule umi_create_whitelist:
    input:
        fastq_dir + &quot;fastq_dir + &quot;SRR1575922_CUT.fastq&quot;,
    output:
        output + &quot;whitelist.txt&quot;
    params:
        bc = &quot;NNNNNN&quot;
    shell:
        &quot;umi_tools whitelist -I {input} --bc-pattern={params.bc} -S {output}&quot;


#add to rule all
#output + &quot;whitelist.txt&quot;

        </code></pre>
<p><strong>Problem- cant use this with no cell indicator</strong> Maybe I dont need this…</p>
<pre><code>rule umi_extract_bc_and_umi:
    input:
        r1 = fastq + &quot;{read_num}_.fastq&quot;,
    output:
        r1_ext = temp(fastq_extr + &quot;combined_r1_extracted.fastq.gz&quot;),
        r2_ext = fastq_extr + &quot;combined_r2_extracted.fastq.gz&quot;
    params:
        bc = barcode
    shell:
        &quot;umi_tools extract --bc-pattern={NNNNNN} --stdin {input.r1} --stdout {output.r1_ext} --read2-in {input.r2} --read2-out={output.r2_ext}  --error-correct-cell --filter-cell-barcode --whitelist={input.wl}&quot;</code></pre>
<p>Rule all:</p>
<pre><code>rule all:
    input:
       fastqc_dir + &quot;SRR1575922_CUT_fastqc.html&quot;,
       dir_genome + ensembl_genome + &quot;.reads&quot;</code></pre>
</div>
<div id="cluster" class="section level3">
<h3>Cluster</h3>
<p>Will need to create a cluster.json file where I specify the memory for each of my rules</p>
<p>Create for first 2 rules:</p>
<pre><code>{
    &quot;__default__&quot; :
    {
        &quot;mem&quot; : 2000,
        &quot;n&quot; : 1,
    &quot;tasks&quot; : 1,
    &quot;name&quot; : &quot;{rule}-{wildcards}&quot;,
    &quot;logfile&quot;: &quot;log/snake-{rule}-{wildcards}-%j.out&quot;
    },

&quot;subread_index&quot; :
    {
            &quot;mem&quot; : 8000
    },
&quot;fastqc&quot; :
    {
        &quot;mem&quot; : 8000
    }
}</code></pre>
<p>Full files are in my pipeline folder on midway</p>
</div>
<div id="run" class="section level3">
<h3>Run</h3>
<p>To run the snakefile, in the pipleline directory <code>scripts/submit-snakemake.sh</code> to check running jobs: <code>squeue -user=brimittleman</code></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.4.2  backports_1.1.1 magrittr_1.5    rprojroot_1.2  
 [5] tools_3.4.2     htmltools_0.3.6 yaml_2.1.14     Rcpp_0.12.13   
 [9] stringi_1.1.5   rmarkdown_1.6   knitr_1.17      git2r_0.19.0   
[13] stringr_1.2.0   digest_0.6.12   evaluate_0.10.1</code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
