<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-04-03</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 0852212</p>
<!-- Add your analysis here -->
<p>I will use this analysis to answer the question: Do eQTLs also equally affect antisense transcriptions? This will take advantage of strand specificity of the net-seq data and the phenominon of convergent and divergent transcription around TSS.</p>
<p>Workflow:</p>
<p>*Download eqtls</p>
<p>*Rank eqtls based on coverage in tss region, this may include seeing if the eqtl genes are in the filtered gene set from the strand specificity analysis</p>
<p>*Look at some of the top snps</p>
<p>-separate individuals by genome<br />
-look at them in IGV<br />
-box plots seperated by genotype, forward and reverse transcription</p>
<div id="load-libraries" class="section level3">
<h3>Load libraries</h3>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>Loading required package: rmarkdown</code></pre>
<pre><code>This is workflowr version 0.7.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyr)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>Warning: package &#39;reshape2&#39; was built under R version 3.4.3</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
</div>
<div id="gene-overlaps" class="section level3">
<h3>Gene overlaps</h3>
<p>Download eqtls and YRI genotypes from <a href="http://eqtl.uchicago.edu/jointLCL/" class="uri">http://eqtl.uchicago.edu/jointLCL/</a> to /project2/gilad/briana/Net-seq-pilot/data/eqtl</p>
<pre class="bash"><code>wget  http://eqtl.uchicago.edu/jointLCL/output_RNAseqGeuvadis_PC14.txt
wget http://eqtl.uchicago.edu/jointLCL/genotypesYRI.gen.txt.gz</code></pre>
<p>Make a list of the genes that have an eqtl with the effect size (so i can order which to look at) :</p>
<pre class="bash"><code>awk &#39;{print $2}&#39; eqtl_RNAseqGeuvadis.txt &gt; eqtl_genes
awk &#39;{print $1 &quot;\t&quot; $2 &quot;\t&quot; $6}&#39; eqtl_RNAseqGeuvadis.txt &gt; eqtl_genes_effectsize.txt</code></pre>
<p>The filtered genes are in /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.bed</p>
<p>Pull both of these into R so I can filter the genes by if they have an eqtl.</p>
<pre class="r"><code>eqtl_gene= read.table(&quot;../data/eqtl_genes_effectsize.txt&quot;, stringsAsFactors = FALSE, header=TRUE)
filter_genes= read.table(&quot;../data/gencode.v19.annotation.distfilteredgenes.bed&quot;, stringsAsFactors = FALSE)
colnames(filter_genes)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;)</code></pre>
<p>Use the filter dplyr functions to filter the eqtl genes by genes that are covered in our list of filtered genes.</p>
<pre class="r"><code>eqtl_gene_filter=eqtl_gene %&gt;% semi_join(filter_genes, by=&quot;gene&quot;) %&gt;% arrange(desc(beta))

eqtl_pergene= eqtl_gene_filter %&gt;% group_by(gene) %&gt;% tally()</code></pre>
<p>This shows that 3985 genes in the filter_genes file have at least one eqtl. The eqtl_gene_filter file has all the eqtl genes sorted in descending order by the beta value. Now I need to back filter the bed file to only include the genes in the eqtl genes. This will allow me to get coverage for each of these.</p>
<pre class="r"><code>filter_genes_byeqtl= filter_genes %&gt;% semi_join(eqtl_pergene, by=&quot;gene&quot;)

#write.table(filter_genes_byeqtl, &quot;../data/gencode.v19.annotation.eqtlfilter.bed&quot;,col.names = TRUE, row.names = FALSE, quote = FALSE, sep=&quot;\t&quot;)</code></pre>
<p>This file is now in /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.eqtlfilter.bed</p>
</div>
<div id="coverage-for-tss" class="section level3">
<h3>Coverage for TSS</h3>
<p>I need to modify gencode.v19.annotation.eqtlfilter.bed to include the TSS of each of these genes. I am looking at the first 500 basepairs of each gene.</p>
<pre class="bash"><code>awk &#39;{if($6 == &quot;+&quot;) print($1 &quot;\t&quot; $2 -500  &quot;\t&quot; $2 + 500  &quot;\t&quot; $4 &quot;\t&quot; $5 &quot;\t&quot; $6 ); else print($1 &quot;\t&quot; $3 - 500 &quot;\t&quot; $3 +500  &quot;\t&quot; $4 &quot;\t&quot; $5 &quot;\t&quot; $6 )}&#39; gencode.v19.annotation.eqtlfilter.bed  &gt; gencode.v19.annotation.eqtlfilter_TSS.bed </code></pre>
<p>Bam to bed and sort each bedfile with sort -k1,1 -k2,2n</p>
<pre class="bash"><code>#!/bin/bash
#SBATCH --mail-type=END

module load Anaconda3
source activate net-seq

bamToBed -i YG-SP-NET3-18508_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-18508_combined_Netpilot-bedsort.bed

bamToBed -i YG-SP-NET3-19128_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-19128_combined_Netpilot-bedsort.bed

bamToBed -i YG-SP-NET3-19141_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-19141_combined_Netpilot-bedsort.bed

bamToBed -i YG-SP-NET3-19193_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-19193_combined_Netpilot-bedsort.bed

bamToBed -i YG-SP-NET3-19239_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-19239_combined_Netpilot-bedsort.bed

bamToBed -i YG-SP-NET3-19257_combined_Netpilot-sort.bam | sort -k1,1 -k2,2n &gt; ../sorted_bed/YG-SP-NET3-19257_combined_Netpilot-bedsort.bed</code></pre>
<p>Now I can write a script that counts the coverage for these TSS in a strand specific manner. Remeber that netseq is flipped strand so same strand should be the -S and opp strand is -s. This script is in the code directory and is called eqtl_stranded.sh.</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=eqtl_strand
#SBATCH --time=8:00:00
#SBATCH --output=eqtlstrand.out
#SBATCH --error=eqtlstrand.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END

module load Anaconda3
source activate net-seq  

sample=$1
describer=$(echo ${sample} | sed -e &#39;s/.*\YG-SP-//&#39; | sed -e &quot;s/_combined_Netpilot-bedsort.bed$//&quot;)


bedtools coverage -counts -S -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.eqtlfilter_TSS.bed -b $1 &gt; /project2/gilad/briana/Net-seq-pilot/output/eqtl_strand_spec/${describer}_same_eqtstrandedlTSS.txt

bedtools coverage -counts -s -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.eqtlfilter_TSS.bed -b $1 &gt; /project2/gilad/briana/Net-seq-pilot/output/eqtl_strand_spec/${describer}_opp_eqtstrandedlTSS.txt
</code></pre>
<p>Create a wrapper to run this on all files: wrap_eqtl_strand.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=w_eqtl_strand
#SBATCH --time=8:00:00
#SBATCH --output=weqtlstrand.out
#SBATCH --error=weqtlstrand.err
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --mail-type=END


for i in $(ls /project2/gilad/briana/Net-seq-pilot/data/sorted_bed/); do
            sbatch eqtl_stranded.sh /project2/gilad/briana/Net-seq-pilot/data/sorted_bed/$i 
        done
        
        
        </code></pre>
<p>need to give more memory to 508, 128, 141, 193, 239</p>
<p>Pull in all files and join them the same strand ones on the first four columns</p>
<pre class="r"><code>same_18486= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18486_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_18486&quot;))
same_18505= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18505_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_18505&quot;))
same_18508= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18508_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_18508&quot;))
same_19128= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19128_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_19128&quot;))
same_19141= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19141_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_19141&quot;))
same_19193= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19193_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_19193&quot;))
same_19239= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19239_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_19239&quot;))
same_19257= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19257_same_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_19257&quot;))</code></pre>
<p>Same strand:</p>
<pre class="r"><code>all_same_strand=cbind(same_18486[,1:7], same_18505$same_18505, same_18508$same_18508, same_19128$same_19128,same_19141$same_19141, same_19193$same_19193, same_19239$same_19239, same_19257$same_19257)
colnames(all_same_strand)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;same_18486&quot;, &quot;same_18505&quot;, &quot;same_18508&quot;, &quot;same_19128&quot;, &quot;same_19141&quot;, &quot;same_19193&quot;, &quot;same_19239&quot;, &quot;same_19257&quot;)</code></pre>
<p>Opp strand:</p>
<pre class="r"><code>opp_18486= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18486_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_18486&quot;))
opp_18505= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18505_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_18505&quot;))
opp_18508= read.table(file = &quot;../data/eqtl_strand_spec/NET3-18508_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_18508&quot;))
opp_19128= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19128_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_19128&quot;))
opp_19141= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19141_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_19141&quot;))
opp_19193= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19193_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_19193&quot;))
opp_19239= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19239_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_19239&quot;))
opp_19257= read.table(file = &quot;../data/eqtl_strand_spec/NET3-19257_opp_eqtstrandedlTSS.txt&quot;, header =F, col.names=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_19257&quot;))</code></pre>
<pre class="r"><code>all_opp_strand=cbind(opp_18486[,1:7], opp_18505$opp_18505, opp_18508$opp_18508, opp_19128$opp_19128,opp_19141$opp_19141, opp_19193$opp_19193, opp_19239$opp_19239, opp_19257$opp_19257)
colnames(all_opp_strand)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;opp_18486&quot;, &quot;opp_18505&quot;, &quot;opp_18508&quot;, &quot;opp_19128&quot;, &quot;opp_19141&quot;, &quot;opp_19193&quot;, &quot;opp_19239&quot;, &quot;opp_19257&quot;)</code></pre>
<p>Calculate the %same:</p>
<pre class="r"><code>perc_18486=all_same_strand$same_18486/(all_same_strand$same_18486 + all_opp_strand$opp_18486)
perc_18505=all_same_strand$same_18505/(all_same_strand$same_18505 + all_opp_strand$opp_18505)
perc_18508=all_same_strand$same_18508/(all_same_strand$same_18508 + all_opp_strand$opp_18508)
perc_19128=all_same_strand$same_19128/(all_same_strand$same_19128 + all_opp_strand$opp_19128)
perc_19141=all_same_strand$same_19141/(all_same_strand$same_19141 + all_opp_strand$opp_19141)
perc_19193=all_same_strand$same_19193/(all_same_strand$same_19193 + all_opp_strand$opp_19193)
perc_19239=all_same_strand$same_19239/(all_same_strand$same_19239 + all_opp_strand$opp_19239)
perc_19257=all_same_strand$same_19257/(all_same_strand$same_19257 + all_opp_strand$opp_19257)</code></pre>
<p>Create a table with these:</p>
<pre class="r"><code>perc_all= cbind(all_same_strand[,1:6],perc_18486, perc_18505,perc_18508,perc_19128,perc_19141,perc_19193,perc_19239,perc_19257)</code></pre>
<p>Make this tidy to remove NA and to make box plots</p>
<pre class="r"><code>perc_all_long=melt(perc_all, variable.name = &quot;library&quot;,  value.name = &quot;perc_same&quot;,
                     id.vars = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;), na.rm = TRUE)</code></pre>
<p>Make a side by side box plot:</p>
<pre class="r"><code>ggplot(perc_all_long, aes(x=library, y=perc_same)) + geom_violin() + ggtitle(&quot;Distribution of the precent of reads mapping to the same strand&quot;) </code></pre>
<p><img src="figure/eqtl_bystrand.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This is good. Expectation is the samples have similar distributions of the</p>
</div>
<div id="session-information" class="section level3">
<h3>Session information</h3>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2    ggplot2_2.2.1   reshape2_1.4.3  dplyr_0.7.4    
[5] tidyr_0.7.2     workflowr_0.7.0 rmarkdown_1.8.5

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15     knitr_1.18       bindr_0.1        magrittr_1.5    
 [5] munsell_0.4.3    colorspace_1.3-2 R6_2.2.2         rlang_0.1.6     
 [9] plyr_1.8.4       stringr_1.2.0    tools_3.4.2      grid_3.4.2      
[13] gtable_0.2.0     git2r_0.21.0     htmltools_0.3.6  lazyeval_0.2.1  
[17] yaml_2.1.16      rprojroot_1.3-2  digest_0.6.14    assertthat_0.2.0
[21] tibble_1.4.2     purrr_0.2.4      glue_1.2.0       evaluate_0.10.1 
[25] labeling_0.3     stringi_1.1.6    compiler_3.4.2   pillar_1.1.0    
[29] scales_0.5.0     backports_1.1.2  pkgconfig_2.0.1 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
