<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p><strong>Last updated:</strong> 2018-07-03</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/Net-seq/tree/06ba7449456fc47bb5e8c9ff36f9dd85802b5cf2" target="_blank">06ba744</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  Ggsb_logo.r.pdf
    Untracked:  Rplot.pdf
    Untracked:  _workflowr.yml
    Untracked:  analysis/filter_bam.Rmd
    Untracked:  analysis/gencode.v19.annotation.proteincodinggene.saf
    Untracked:  analysis/temp
    Untracked:  analysis/top5_gen_wind200.bed
    Untracked:  data/DaPars_APA_geuvadis.txt
    Untracked:  data/Day7_cardiomyocytes_droNC_seq.bam
    Untracked:  data/Day7_cardiomyocytes_droNC_seq.bam.bai
    Untracked:  data/Day7_cardiomyocytes_drop_seq.bam
    Untracked:  data/Day7_cardiomyocytes_drop_seq.bam.bai
    Untracked:  data/LCL_3utr.txt
    Untracked:  data/LCL_3utrAB.bed
    Untracked:  data/LCL_3utrAB.neg.chr20.bed
    Untracked:  data/LCL_3utrAB_pos.chr1.bed
    Untracked:  data/LCL_3utrAB_pos.chr21.bed
    Untracked:  data/NET3-18486.gene.coverage.bed
    Untracked:  data/NET3-18486.gene.coverage.noSM.bed
    Untracked:  data/NET3-18486.gene.coverage.nosn.nosno.bed
    Untracked:  data/NET3-18486.gene.coverage.notopwind.bed
    Untracked:  data/NET3-18486.tss.coverage.bed
    Untracked:  data/NET3-18486_combined_Netpilot-sort.FC200.cov.bed
    Untracked:  data/NET3-18486_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-18486_combined_Netpilot-sort.exon.cov.txt
    Untracked:  data/NET3-18505.gene.coverage.bed
    Untracked:  data/NET3-18505_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-18508_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-19128_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-19141_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-19193_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-19239_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/NET3-19257_combined_Netpilot-sort.FC200.cov.no0.bed
    Untracked:  data/RNAseqGeuvadis_STAR_18486.coverage.bed
    Untracked:  data/RNAseqGeuvadis_STAR_18486.gene.coverage.bed
    Untracked:  data/RefSeqGenes.bed
    Untracked:  data/SRR1575922-sort.bam
    Untracked:  data/SRR1575922-sort.bam.bai
    Untracked:  data/SwitchGear_TSS.bed
    Untracked:  data/UMI_18486_dep_stat.txt
    Untracked:  data/UMI_18486_dep_stat_tab.txt
    Untracked:  data/UMI_18508_dep_stat.txt
    Untracked:  data/UMI_18508_nondep_stat.txt
    Untracked:  data/UMI_19238_dep_stat.txt
    Untracked:  data/UMI_Net3_18486_dedupstat.txt
    Untracked:  data/UMI_Net3_18486_stat.txt
    Untracked:  data/UMI_Net3_18505_dedupstat.txt
    Untracked:  data/UMI_Net3_18505_stat.txt
    Untracked:  data/UMI_Net3_18508_dedupstat.txt
    Untracked:  data/UMI_Net3_18508_stat.txt
    Untracked:  data/UMI_Net3_19128_dedupstat.txt
    Untracked:  data/UMI_Net3_19128_stat.txt
    Untracked:  data/UMI_Net3_19141_dedupstat.txt
    Untracked:  data/UMI_Net3_19141_stat.txt
    Untracked:  data/UMI_Net3_19193_dedupstat.txt
    Untracked:  data/UMI_Net3_19193_stat.txt
    Untracked:  data/UMI_Net3_19239_dedupstat.txt
    Untracked:  data/UMI_Net3_19239_stat.txt
    Untracked:  data/UMI_Net3_19257_dedupstat.txt
    Untracked:  data/UMI_Net3_19257_stat.txt
    Untracked:  data/UMI_mayer_stat.txt
    Untracked:  data/YG-SP-NET1-18486-dep-2017-10-13_S4_R1_001-sort.dedup.cov.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.AC093901.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.BTRC.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.RNU5B.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.WDR74.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.chr2.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.insig2.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.ppef2.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.cov.rnu259p.bed
    Untracked:  data/YG-SP-NET3-18486_combined_Netpilot-sort.dedup.cov.insig2.bed
    Untracked:  data/all_RNAmetrics.picard.none.csv
    Untracked:  data/all_files.APA500.coverage.bed
    Untracked:  data/all_files_coverage.bed
    Untracked:  data/all_strand_genecounts_18486.txt
    Untracked:  data/bam_files_chr/
    Untracked:  data/blcl.hg38.sorted.bam
    Untracked:  data/blcl.hg38.sorted.bam.bai
    Untracked:  data/cell_growth_3.21.18.csv
    Untracked:  data/clip_18486_dep.txt
    Untracked:  data/clusters.bed
    Untracked:  data/clusters.hg38
    Untracked:  data/clusters.hg38.3utr.neg.bed
    Untracked:  data/clusters.hg38.3utr.pos.bed
    Untracked:  data/clusters.hg38.bed
    Untracked:  data/dedup_18486_mapqual.txt
    Untracked:  data/drop7_cardio_3utr.txt
    Untracked:  data/drop7_cardio_3utrAB.neg.chr21.bed
    Untracked:  data/drop7_cardio_3utrAB.pos.chr21.bed
    Untracked:  data/ensembl2refseq.txt
    Untracked:  data/eqtl_fullgene/
    Untracked:  data/eqtl_genes_effectsize.txt
    Untracked:  data/eqtl_output.cis.txt
    Untracked:  data/eqtl_output.txt
    Untracked:  data/eqtl_strand_spec/
    Untracked:  data/exon_cov/
    Untracked:  data/fc_genecov/
    Untracked:  data/gencode.v19.annotation.distfilteredgenes.bed
    Untracked:  data/gencode.v19.annotation.egqtlfilter.bed
    Untracked:  data/gencode.v19.annotation.eqtlfilter.bed
    Untracked:  data/gencov_18486.bed
    Untracked:  data/gene_cov_count/
    Untracked:  data/gene_coverage_18486_dedup_hist.txt
    Untracked:  data/gene_coverage_18486_hist.txt
    Untracked:  data/gene_coverage_18508_dep.txt
    Untracked:  data/gene_coverage_18508_dep_hist.txt
    Untracked:  data/gene_coverage_18508_nondep_hist.txt
    Untracked:  data/gene_coverage_19238_dep_hist.txt
    Untracked:  data/gene_coverage_mayer_SRR1575922_hist.txt
    Untracked:  data/gene_dedup_cov_count/
    Untracked:  data/genotypes.rs7144811.txt
    Untracked:  data/growth_curve_3.16.csv
    Untracked:  data/hES.hg38.sorted.bam
    Untracked:  data/hES.hg38.sorted.bam.bai
    Untracked:  data/hg19.GM72.CTCF
    Untracked:  data/hg19.ref.genes.bed
    Untracked:  data/insig2sec.txt
    Untracked:  data/mapped_18486_dep.txt
    Untracked:  data/mapped_18486_dep_max.txt
    Untracked:  data/mapped_18508_dep.txt
    Untracked:  data/mapped_19238_dep.txt
    Untracked:  data/mapped_mayer.txt
    Untracked:  data/mapped_qual_18486.txt
    Untracked:  data/mapped_qual_18505.txt
    Untracked:  data/mapped_qual_18508.txt
    Untracked:  data/mapped_qual_19128.txt
    Untracked:  data/mapped_qual_19141.txt
    Untracked:  data/mapped_qual_19193.txt
    Untracked:  data/mapped_qual_19239.txt
    Untracked:  data/mapped_qual_19257.txt
    Untracked:  data/matrix_expression.txt
    Untracked:  data/matrix_genotypes.csv
    Untracked:  data/matrix_genotypes.txt
    Untracked:  data/merged_Net1.bam
    Untracked:  data/merged_Net1.bam.bai
    Untracked:  data/meta_info_coverage.bed
    Untracked:  data/names_geno.txt
    Untracked:  data/net-3-readmap/
    Untracked:  data/net1_18486_dep_dedup.bed
    Untracked:  data/net1_18486_dep_dedup_chr.bed
    Untracked:  data/net4_readcounts.xlsx
    Untracked:  data/net_pilot_eqtl_expression.bed
    Untracked:  data/net_pilot_eqtl_genotypes.vcf
    Untracked:  data/netcomb_intronicbases.csv
    Untracked:  data/opp_strand_genecounts_18486.txt
    Untracked:  data/opp_strand_genecounts_18505.txt
    Untracked:  data/opp_strand_genecounts_filt_18486.txt
    Untracked:  data/perc_alive_3.16.csv
    Untracked:  data/prom_coverage/
    Untracked:  data/qual_18486_dep.txt
    Untracked:  data/qual_18508_dep.txt
    Untracked:  data/qual_19238_dep.txt
    Untracked:  data/qual_mayer.txt
    Untracked:  data/refseq_250up.bed
    Untracked:  data/run_lm_APA.txt
    Untracked:  data/same_strand_genecounts_18486.txt
    Untracked:  data/same_strand_genecounts_18505.txt
    Untracked:  data/same_strand_genecounts_filt_18486.txt
    Untracked:  data/sort_dedup_3prime_chr2_no0.18486.txt
    Untracked:  data/sort_dedup_chr2_no0_18486.txt
    Untracked:  data/test.txt
    Untracked:  data/three_prime_utr.bed
    Untracked:  data/top5_exonlist.txt
    Untracked:  data/top5_exonlist_18486_fiveprime_cov.txt
    Untracked:  data/top5_exonlist_18486_fiveprime_cov2.txt
    Untracked:  data/top5_exonlist_18486_fiveprime_cov2_filter.txt
    Untracked:  data/top5_exonlist_18486_threeprime_cov.txt
    Untracked:  data/top5_exonlist_18486_threeprime_cov2.txt
    Untracked:  data/top5_exonlist_18486_threeprime_cov2_filter.txt
    Untracked:  data/top5_gen_wind200.bed
    Untracked:  data/top5_gen_wind200.tab.bed
    Untracked:  data/uniq_genes/
    Untracked:  data/windows_200/
    Untracked:  docs/temp
    Untracked:  docs/top5_gen_wind200.bed
    Untracked:  output/Rs7144811_apa_usage.pdf
    Untracked:  output/picard.accrossgenebodies.netpilot.csv

Unstaged changes:
    Modified:   analysis/APA_qtl_RNAseq.Rmd
    Modified:   analysis/LCL_growth.Rmd
    Modified:   analysis/Net_3_explore.Rmd
    Modified:   analysis/Reads_per_pas.Rmd
    Modified:   analysis/UTR_coverage.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/bin_windows.Rmd
    Modified:   analysis/check_bamid.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/conda.environment.Rmd
    Modified:   analysis/config.snake.setup.Rmd
    Modified:   analysis/create_blacklist.Rmd
    Modified:   analysis/data_for_ggplot.Rmd
    Modified:   analysis/eqtl_bystrand.Rmd
    Modified:   analysis/explore_umi_usage.Rmd
    Modified:   analysis/extend_APA_qtl.Rmd
    Modified:   analysis/gviz_plots.Rmd
    Modified:   analysis/initial.data.exploration.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/map_stats_from_bam.Rmd
    Modified:   analysis/reads_in_genes.Rmd
    Modified:   analysis/recreate_mayer_figs.Rmd
    Modified:   analysis/strand_spec.Rmd
    Modified:   analysis/test-analysis.Rmd
    Modified:   analysis/three_prime_UTR.Rmd
    Modified:   analysis/update_snakefile.Rmd
    Modified:   analysis/use_deeptools.Rmd
    Modified:   analysis/visualize_genomefeatures.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/363ea13cbf94c567f793607bb0d41a615dd4f496/docs/index.html" target="_blank">363ea13</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-22
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/97a0c53cbc5fb104ac675c613ba1be3b296e7923/analysis/index.Rmd" target="_blank">97a0c53</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-22
</td>
<td style="text-align:left;">
wflow_publish(c(“analysis/index.Rmd”, “analysis/net-4-explore.Rmd”))
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/d76f54df88e5a906e6dc91dbceac779a07c1680f/docs/index.html" target="_blank">d76f54d</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-04-05
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/c269f0fbfb679f720ac6de4910f8f8026c58a722/analysis/index.Rmd" target="_blank">c269f0f</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-04-05
</td>
<td style="text-align:left;">
fixr eqtl by strand bullet
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/465e6160ccb3a7b06968b700a294d00566840779/docs/index.html" target="_blank">465e616</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-04-03
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/0a0b388d885ae673a8aacbbc278e2e137afb3502/docs/index.html" target="_blank">0a0b388</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-22
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/0e5035278ba028867eb9b14678a7dfda8936ff82/analysis/index.Rmd" target="_blank">0e50352</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-22
</td>
<td style="text-align:left;">
start strand specific analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/09e78cf51b8caada6ff7abaeffc926ff21caa068/docs/index.html" target="_blank">09e78cf</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-19
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/3aa98a805a83c0a25da1020064d966bc2d8a9119/analysis/index.Rmd" target="_blank">3aa98a8</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-19
</td>
<td style="text-align:left;">
add recreate rmd to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/fd42b704d21173499035f47a69a4bfe7fda4d756/docs/index.html" target="_blank">fd42b70</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/1707538feee1b758dec66e37a61b5affbcabdd13/analysis/index.Rmd" target="_blank">1707538</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-16
</td>
<td style="text-align:left;">
initialize growth analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/b789dd901188ced31d2f53bd7ed5b01dd3144f59/docs/index.html" target="_blank">b789dd9</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-02
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/646ee57d2ca74adb87cd16cbfd943147d7492f9c/analysis/index.Rmd" target="_blank">646ee57</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-03-02
</td>
<td style="text-align:left;">
initialize analysis for the pilot data
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/a84f24aa21c9d306b81ad2e97136db1261c68af5/docs/index.html" target="_blank">a84f24a</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-02-14
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/d72c4479f76fe7f4aa475e25ef4dd1b93446e754/docs/index.html" target="_blank">d72c447</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-02-02
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/01f846fd30e3e6eeabe8c8ebd613c6c72db11b72/analysis/index.Rmd" target="_blank">01f846f</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-02-02
</td>
<td style="text-align:left;">
wflow_publish(c(“index.Rmd”, “use_deeptools.Rmd”))
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/558ff5af77b5ae43df26f4458f35c75134990d48/docs/index.html" target="_blank">558ff5a</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-01-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/308d2b59def93235905d7822c7da2d1d16e7b869/analysis/index.Rmd" target="_blank">308d2b5</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-01-31
</td>
<td style="text-align:left;">
add utr cov to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/2f36a96770f827edce9e3f8e42d7836550f9a4c4/docs/index.html" target="_blank">2f36a96</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/52df489ec00b093bab48c5c8453460dc4a942988/analysis/index.Rmd" target="_blank">52df489</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-01-25
</td>
<td style="text-align:left;">
stranded graphs and add to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/8a554c1ff315190fb4146ccc6e7548ab00eae5d9/docs/index.html" target="_blank">8a554c1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-12-13
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/3ed7074f3156c2e55165b045d8f6959cc1196f6d/analysis/index.Rmd" target="_blank">3ed7074</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-12-13
</td>
<td style="text-align:left;">
create plots for actb and add to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/3326147fba586f15e436a748802292a1417dee05/docs/index.html" target="_blank">3326147</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-12-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/fab79cab4fcd5315bddb6350701a65e3882d5cce/analysis/index.Rmd" target="_blank">fab79ca</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-12-08
</td>
<td style="text-align:left;">
reforat gene count data for ggplot
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/f006037a1cd1c5c2b2e1bb16498fa1300f4e4ea4/docs/index.html" target="_blank">f006037</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/b180d9a01198c8c139e3f238b956e64e39d59f55/analysis/index.Rmd" target="_blank">b180d9a</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-30
</td>
<td style="text-align:left;">
fix index formating
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/ec2d994d682ddfdd0d8c8157e062a074d653225d/docs/index.html" target="_blank">ec2d994</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/35b1aa1f5e09c98f20178178d8d01d53212bf1c5/analysis/index.Rmd" target="_blank">35b1aa1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-30
</td>
<td style="text-align:left;">
wflow_publish(c(“bin_windows.Rmd”, “index.Rmd”, “reads_in_genes.Rmd”))
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/552910dcb689147e6289bc0eba0d4a514de43bcd/docs/index.html" target="_blank">552910d</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-28
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/7041a545e691fe2c6dab16451b94bea919c7913f/analysis/index.Rmd" target="_blank">7041a54</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-28
</td>
<td style="text-align:left;">
add reads_in_gene
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/697f6a0f05e272a6193455465ef3466dfbdbe5d3/docs/index.html" target="_blank">697f6a0</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-20
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/560df7decda675e2b2dc88444e60e5f75566d3e9/analysis/index.Rmd" target="_blank">560df7d</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-20
</td>
<td style="text-align:left;">
initiate the reads in gene analysis and add it to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/121c3e04d637c1b26135a38e0fd409b253cae447/docs/index.html" target="_blank">121c3e0</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/acd27cd6d5f49fce1b125b260890ec69a16967b5/analysis/index.Rmd" target="_blank">acd27cd</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-16
</td>
<td style="text-align:left;">
add map_stats to inde
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/3d6ebfe7d63b71665af72a2bababdba2fc83fee2/docs/index.html" target="_blank">3d6ebfe</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-09
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/aee82258fc88ef2fc6ac604c4816d0247a98598d/analysis/index.Rmd" target="_blank">aee8225</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-09
</td>
<td style="text-align:left;">
initialize and load data for genomation data
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/1d36decff76c9db2cbb60e352c7852c3855c427b/docs/index.html" target="_blank">1d36dec</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/9a86ba3b32d594c518b0199d30f7785e6d9ec0e0/analysis/index.Rmd" target="_blank">9a86ba3</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-08
</td>
<td style="text-align:left;">
Initiate the explore UMI usage analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/4207b6056059bd8788bdeaec91ebbff63770071f/docs/index.html" target="_blank">4207b60</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-06
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/7eeb4e16239b5632a94a51e831c8aeb62473588d/analysis/index.Rmd" target="_blank">7eeb4e1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-11-06
</td>
<td style="text-align:left;">
start initial data exploration and add it to index
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/baeec88f52f0c7e11cef08b9e71f32f834a0fda5/analysis/index.Rmd" target="_blank">baeec88</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-26
</td>
<td style="text-align:left;">
commit index and config changes
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/5096c149196360f8fb23b0f26ae48ff61a076b38/docs/index.html" target="_blank">5096c14</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-26
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/4c4559226b7685c55ad517429e2162617dfa046b/analysis/index.Rmd" target="_blank">4c45592</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-26
</td>
<td style="text-align:left;">
add new config/snake file to index
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/0ea88ccf4323ab0e153ae91faf74e5bd9f339dbb/docs/index.html" target="_blank">0ea88cc</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-24
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/13f7b6ac73c9578414dbadc8b4b675d1f98af287/analysis/index.Rmd" target="_blank">13f7b6a</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-24
</td>
<td style="text-align:left;">
move conda code to conda environment analysis and update
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/5f7df97223d79c058f3bf8bfcb195d3b37c1a1f1/docs/index.html" target="_blank">5f7df97</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-24
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/f03ffb4b5c727f92c4411600488c6f6da773f2f6/analysis/index.Rmd" target="_blank">f03ffb4</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-17
</td>
<td style="text-align:left;">
add test analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/e445d99443196f44f631359d2d8a209fa9c5ba79/docs/index.html" target="_blank">e445d99</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-17
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/Net-seq/6ee857dc16959b0fc43ccd23e7124c9cbba5d4cd/docs/index.html" target="_blank">6ee857d</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-17
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/c88cd2e2ba105bf22d9cea7a6cc81f586d3eddca/analysis/index.Rmd" target="_blank">c88cd2e</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-17
</td>
<td style="text-align:left;">
Publish the initial files for myproject
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/Net-seq/blob/d63924864b8a1e3784190f0ee100a543321f47c2/analysis/index.Rmd" target="_blank">d639248</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2017-10-17
</td>
<td style="text-align:left;">
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>This is the workflowr site for my Net-seq project. I will have all of the data, code, and analysis here.</p>
<p>Set-up for mappping my libraries:</p>
<ul>
<li><p><a href="conda.environment.html">Setting up conda environment</a></p></li>
<li><p><a href="config.snake.setup.html">Setting up config and snake files</a></p></li>
</ul>
<p>Initial data exploration:</p>
<ul>
<li><p><a href="initial.data.exploration.html">Initial data exploration for NetSeq1 libary</a></p></li>
<li><p><a href="map_stats_from_bam.html">Explore read lengths that are mapping</a></p></li>
<li><p><a href="explore_umi_usage.html">Explore the UMI usage in the NetSeq1 library</a></p></li>
<li><p><a href="visualize_genomefeatures.html">Use the genomation package to vizualize genome features and see overlap with Net-seq1</a></p></li>
<li><p><a href="reads_in_genes.html">Exploration of read counts per gene</a></p></li>
<li><p><a href="data_for_ggplot.html">Reformat gene counts for ggplot2 data analysis</a></p></li>
<li><p><a href="bin_windows.html">Explore read counts per genome bin</a></p></li>
<li><p><a href="gviz_plots.html">Create IGV browser style plots</a></p></li>
</ul>
<p>Initial analysis for Net-3:</p>
<ul>
<li><a href="Net_3_explore.html">Net-3 Initial exploration</a></li>
</ul>
<p>Net-pilot analysis:</p>
<ul>
<li><p><a href="update_snakefile.html">Update snakefile for pilot analysis</a></p></li>
<li><p><a href="create_blacklist.html">Create blacklist to filter reads</a></p></li>
<li><p><a href="recreate_mayer_figs.html">Recreate figures from Mayer</a></p></li>
<li><p><a href="strand_spec.html">Strand specific analysis</a></p></li>
<li><p><a href="eqtl_bystrand.html">Connect eQTLs to strand</a></p></li>
</ul>
<p>Cell Growth analysis:</p>
<ul>
<li><a href="LCL_growth.html">Initial plots on growth</a></li>
</ul>
<p>Exploration of the 3` seq and ApA sites:</p>
<ul>
<li><p><a href="three_prime_UTR.html">Look for read enrichment at ApA sites</a></p></li>
<li><p><a href="UTR_coverage.html">UTR coverage analysis</a></p></li>
<li><p><a href="use_deeptools.html">Look at enrichment using Deep tools</a></p></li>
</ul>
<p>Net 4 analysis- includes 3 lanes for all 16 individuals:</p>
<ul>
<li><a href="net-4-explore.Rmd">Explore Net 4</a></li>
</ul>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
