<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-02-14" />

<title>Initial Data exploration of Net-3</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Initial Data exploration of Net-3</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2018-02-14</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-02-18</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> c58a151</p>
<!-- Add your analysis here -->
<div id="mapped-reads" class="section level2">
<h2>Mapped reads</h2>
<p>This analysis is to look at the quality of the library net-3. This has the first 8 libraries from the 16 pilot libraries.</p>
<ul>
<li><p>18505</p></li>
<li><p>18508</p></li>
<li><p>18486</p></li>
<li><p>19239</p></li>
<li><p>19141</p></li>
<li><p>19193</p></li>
<li><p>19257</p></li>
<li><p>19128</p></li>
</ul>
<p>The first step is to run the raw reads through my snakefile. To do this I have to change the project directory in the /project2/gilad/briana/net_seq_pipeline/config.yaml to /project2/gilad/briana/Net-seq/Net-seq3/ .</p>
<p>The reads will go in: /project2/gilad/briana/Net-seq/Net-seq3/data/fastq/</p>
<p>I first use gunzip to unzip all of the fastq files then I submit the snakemake file from /project2/gilad/briana/net_seq_pipeline with nohup scripts/submit-snakemake.sh</p>
<p>Look at number of reads, mapped reads, deduplicated reads.</p>
<p>‘samtools view -c -F 4’</p>
<pre class="r"><code>libraries=c(&quot;18486&quot;, &quot;18505&quot;, &quot;18508&quot;, &quot;19128&quot;, &quot;19141&quot;, &quot;19193&quot;, &quot;19239&quot;, &#39;19257&#39;)
fastq=c( 34342214,42959246,205654644, 51413661,211367408,52841852,54253650,45506210)
mapped=c(14018943,23607050,26532568,38084941,28165596,37084191,37095436,26636660)
dedup= c(1446379, 1032174,1642314,2469438,1926300,2717515, 2361238, 1377712 )

read_count=rbind(fastq,mapped,dedup)
colnames(read_count)=libraries </code></pre>
<pre class="r"><code>total_reads= sum(fastq)
total_reads</code></pre>
<pre><code>[1] 698338885</code></pre>
<pre class="r"><code>total_map= sum(mapped)
total_map</code></pre>
<pre><code>[1] 231225385</code></pre>
<pre class="r"><code>total_dedup= sum(dedup)
total_dedup</code></pre>
<pre><code>[1] 14973070</code></pre>
<pre class="r"><code>total_map/total_reads</code></pre>
<pre><code>[1] 0.3311077</code></pre>
<pre class="r"><code>total_dedup/total_map</code></pre>
<pre><code>[1] 0.0647553</code></pre>
<p>Make a plot to look at this by line:</p>
<pre class="r"><code>count_plot=barplot(as.matrix(read_count), main=&quot;Counts for coverage and complexity&quot;,
  xlab=&quot;Library&quot;, col=c(&quot;lightskyblue2&quot;,&quot;dodgerblue1&quot;,&quot;navy&quot;),
    ylab=&quot;Read counts&quot;)
legend(&quot;topright&quot;, legend = c(&quot;total&quot;, &quot;mapped&quot;, &quot;UMI&quot;), col=c(&quot;lightskyblue2&quot;,&quot;dodgerblue1&quot;,&quot;navy&quot;), pch=20, cex = .75)</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Look at this by proportions:</p>
<pre class="r"><code>percent_mapped= mapped/fastq
percent_UMI= dedup/fastq
percent_mapped= percent_mapped - percent_UMI
percent_not_mapped= 1- percent_mapped - percent_UMI


prop=rbind(percent_not_mapped, percent_mapped, percent_UMI)
colnames(prop)= libraries

prop_plot=barplot(as.matrix(prop), main=&quot;Proportions for coverage and complexity&quot;,
  xlab=&quot;Library&quot;, col=c(&quot;lightskyblue2&quot;,&quot;dodgerblue1&quot;,&quot;navy&quot;),
    ylab=&quot;Proportion of sequenced reads&quot;)
legend(&quot;bottomright&quot;, legend = c(&quot;un-mapped&quot;, &quot;mapped&quot;, &quot;UMI&quot;), col=c(&quot;lightskyblue2&quot;,&quot;dodgerblue1&quot;,&quot;navy&quot;), pch=20, cex = 0.75)</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(mapped)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
14018943 25801188 27401128 28903173 37087002 38084941 </code></pre>
<pre class="r"><code>summary(percent_mapped)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.1210  0.3056  0.5403  0.4594  0.6428  0.6927 </code></pre>
<pre class="r"><code>summary(percent_not_mapped)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2592  0.3117  0.4326  0.5085  0.6605  0.8710 </code></pre>
<p>I will now use the get_mapped.py script from the previous analysis to see how long the mapped reads are. I copied it to /project2/gilad/briana/Net-seq/Net-seq3/code. The code takes in sorted bam file and an output text file. I will create a sbatch script to run this on all 8 files. This file is called wrap_get_mapped.sh and is in the same code directory.</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=wrap_get_map
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3

source activate net-seq  

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_18486.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18505-2018-1-8_S1_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_18505.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18508-2018-1-8_S2_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_18508.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19128-2018-1-8_S8_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_19128.txt  


python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19141-2018-1-8_S5_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_19141.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19193-2018-1-8_S6_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_19193.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19239-2018-1-8_S4_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_19239.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_mapped.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19257-2018-1-8_S7_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_stats_19257.txt 
</code></pre>
<p>I will now make histograms showing the number of reads that map from the cigar strings.</p>
<pre class="r"><code>#pull in data:  

map.vec.18486=read.csv(&quot;../data/net-3-readmap/mapped_stats_18486.txt&quot;, header=FALSE)
map.vec.18505=read.csv(&quot;../data/net-3-readmap/mapped_stats_18505.txt&quot;, header=FALSE)
map.vec.18508=read.csv(&quot;../data/net-3-readmap/mapped_stats_18508.txt&quot;, header=FALSE)
map.vec.19128=read.csv(&quot;../data/net-3-readmap/mapped_stats_19128.txt&quot;, header=FALSE)
map.vec.19141=read.csv(&quot;../data/net-3-readmap/mapped_stats_19141.txt&quot;, header=FALSE)
map.vec.19193=read.csv(&quot;../data/net-3-readmap/mapped_stats_19193.txt&quot;, header=FALSE)
map.vec.19239=read.csv(&quot;../data/net-3-readmap/mapped_stats_19239.txt&quot;, header=FALSE)
map.vec.19257=read.csv(&quot;../data/net-3-readmap/mapped_stats_19257.txt&quot;, header=FALSE)
#pull in mayer as well

map.vec.mayer=read.csv(&quot;../data/mapped_mayer.txt&quot;, header=FALSE)</code></pre>
<pre class="r"><code>#make histograms

par(mfrow=c(3,3))
hist((as.numeric(map.vec.18486$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 18486 mapped reads&quot;)
 hist((as.numeric(map.vec.18505$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 18505 mapped reads&quot;)
hist((as.numeric(map.vec.18508$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 18508 mapped reads&quot;)
hist((as.numeric(map.vec.19128$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 19128 mapped reads&quot;)
hist((as.numeric(map.vec.19141$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 19141 mapped reads&quot;)
hist((as.numeric(map.vec.19193$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 19193 mapped reads&quot;)
hist((as.numeric(map.vec.19239$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 19239 mapped reads&quot;)
hist((as.numeric(map.vec.19257$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of 19257 mapped reads&quot;)
hist((as.numeric(map.vec.mayer$V1)), xlab=&quot;Bases&quot;, main=&quot;Histogram of Mayer mapped reads&quot;)</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="deep-tools-analysis" class="section level2">
<h2>Deep tools analysis:</h2>
<p>First I downloaded the gencode.hg19 gene gtf file. I will use bedops convert2bed.</p>
<pre class="bash"><code>awk &#39;{ if ($0 ~ &quot;transcript_id&quot;) print $0; else print $0&quot; transcript_id \&quot;\&quot;;&quot;; }&#39; gencode.v19.annotation.gtf | gtf2bed - &gt; gencode.v19.annotation.bed 
</code></pre>
<p>Deep tools code:</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=deeptools_netseq
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END
#SBATCH --output=deeptool_sbatch.out
#SBATCH --error=deeptools_sbatch.err

module load Anaconda3

source activate net-seq

bamCoverage -b /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam -o /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.bw

computeMatrix reference-point -S /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.bw -R /project2/gilad/briana/Net-seq/Net-seq3/gencode.v19.annotation.bed --referencePoint TSS -b 100 -a 100 -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.gz

plotHeatmap --sortRegions descend --refPointLabel &quot;TSS&quot; -m /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.gz  -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.png</code></pre>
<p>To look at TES I will change the reference point to TES. This script is also in the code file as deep_tools_TES_net18486.sh.</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=deeptools_TES_netseq
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END
#SBATCH --output=deeptool_TES_sbatch.out
#SBATCH --error=deeptools_TES_sbatch.err

module load Anaconda3

source activate net-seq

#the bw file has already been created

computeMatrix reference-point -S /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.bw -R /project2/gilad/briana/Net-seq/Net-seq3/gencode.v19.annotation.bed --referencePoint TES -b 100o -a 1000 -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-TES-18486.gz

plotHeatmap --sortRegions descend --refPointLabel &quot;TSS&quot;  -m /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-TES-18486.gz  -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-TES-18486.png</code></pre>
<p>Do this for DNA hypersensitivity clusters as well. This script is in the code file as deep_tools_DHS_net18486.sh.</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=deeptools_DHS_netseq
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END
#SBATCH --output=deeptool_DHS_sbatch.out
#SBATCH --error=deeptools_DHS_sbatch.err

module load Anaconda3

source activate net-seq

#the bw file has already been created

computeMatrix reference-point -S /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-18486.bw -R /project2/gilad/briana/Net-seq/Net-seq3/wgEncodeRegDnaseClusteredV3.bed -b 1000 -a 1000 -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-dhs-18486.gz

plotHeatmap --sortRegions descend --refPointLabel &quot;DHS&quot; -m /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-dhs-18486.gz  -out /project2/gilad/briana/Net-seq/Net-seq3/output/deeptools/net-seq-dhs-18486.png
</code></pre>
<p>I am running these and 1 more looking at the TSS but at 1000 base pairs on each side.<br />
<strong>timed out at 1000 need to try 500 on each side</strong></p>
</div>
<div id="gene-counts-with-feature-counts" class="section level2">
<h2>Gene counts with Feature counts</h2>
<p>I will use featureCounts to count the number of reads at each gene in the gencode.v19.annotation.gtf file. I will make a file that takes the bam as a input so I can run it on each bam file.</p>
<p>‘Usage: featureCounts [options] -a <annotation_file> -o <output_file> input_file1’</p>
<p>This file is in the net3 code directory. featureCounts_genes.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=featureCounts
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END
#SBATCH --output=featurecount.out
#SBATCH --error=featurecount.err


#$1 input bam full file  

module load Anaconda3

source activate net-seq


sample=$1


describer=$(echo ${sample} | sed -e &#39;s/.*\YG-SP-//&#39; | sed -e &quot;s/_L008_R1_001-sort.bam$//&quot;)


featureCounts -a /project2/gilad/briana/Net-seq/Net-seq3/gencode.v19.annotation.gtf -o /project2/gilad/briana/Net-seq/Net-seq3/output/featureCounts/${sample}-gene-counts.txt ${sample}
</code></pre>
<p>Try this with one file. /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam</p>
<p>Bedtools coverage is a better program for this analyisis. I want to look at coverage along the full gene, not by exon/transcript.</p>
<p>Need to cut the chr from the gencode file so this matches the way my bam files are written. <code>cat gencode.Genes.bed |sed 's/^chr//' &gt; gencode_noCHR_genes.bed</code></p>
<p>This is in the code directory as gencode_cov.sh</p>
<pre class="bash"><code>
#!/bin/bash


#SBATCH --job-name=count_cov
#SBATCH --output=count_cov_sbatch.out
#SBATCH --error=count_cov_sbatch.err
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3  

source activate net-seq 


sample=$1

describer=$(echo ${sample} | sed -e &#39;s/.*\YG-SP-//&#39; | sed -e &quot;s/_L008_R1_001-sort.bam$//&quot;)

bedtools coverage -counts -sorted -a /project2/gilad/briana/Net-seq/Net-seq3/gencode_noCHR_genes_MT_Fsort.bed -b $1 &gt; /project2/gilad/briana/Net-seq/Net-seq3/data/gencode_cov/${describer}.coverage.bed
</code></pre>
<p>run on /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam</p>
<p>I will run this on all 8 files then merge the files.</p>
<p>To merge:</p>
<pre class="bash"><code>
touch all_files_coverage.bed
awk &#39;{print $7}&#39; NET3-18486-2018-1-8_S3.coverage.bed | paste -d&#39;&#39; all_files_coverage.bed  &gt; all_files_coverage.bed


cp all_files_coverage.bed tmp 
for i in `cat list_of_files.txt`; do awk &#39;NR==FNR {a[NR]=$7;next} {print $0,a[FNR]}&#39; $i tmp &gt; tmp2; mv tmp2 tmp; done

cut -f -6  NET3-18486-2018-1-8_S3.coverage.bed &gt; meta_info_coverage.bed
</code></pre>
<p>Now I will have the meta data seperate from the matrix of counts.</p>
<pre class="r"><code>net_coverage=read.table(&quot;../data/all_files_coverage.bed&quot;, header=FALSE)
colnames(net_coverage)= libraries
net_coverage_meta=read.table(&quot;../data/meta_info_coverage.bed&quot;, header=FALSE)
colnames(net_coverage_meta)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;)</code></pre>
<p>Make one with counts and meta info:</p>
<pre class="r"><code>net_cov_full= cbind(net_coverage_meta, net_coverage)</code></pre>
<p>Make a function that will tell me proportion of genes detected.</p>
<pre class="r"><code>genes_detected=function(col){
  #takes in net_cov_full col which corresponds to one library
  tot_genes=nrow(net_cov_full)
  exp_genes=sum(col!=0)
  return(exp_genes/tot_genes)
}

prop_non0= c(genes_detected(net_cov_full$`18486`), genes_detected(net_cov_full$`18505`),genes_detected(net_cov_full$`18508`), genes_detected(net_cov_full$`19128`), genes_detected(net_cov_full$`19141`), genes_detected(net_cov_full$`19193`), genes_detected(net_cov_full$`19239`), genes_detected(net_cov_full$`19257`))
names(prop_non0)=libraries

barplot(prop_non0, ylim = c(0,1), main=&quot;Net-seq Genes detected&quot;, ylab=&quot;Proportion non zero genes&quot;, xlab=&quot;Library&quot;, col = &#39;Blue&#39;)</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Heatmap to see relatedness:</p>
<pre class="r"><code>cor_function=function(data){
  corr_matrix= matrix(0,8,8)
  for (i in seq(1,8)){
    for (j in seq(1,8)){
      x=cor.test(net_coverage[,i],  net_coverage[,j], method=&#39;pearson&#39;)
      cor_ij=as.numeric(x$estimate)
      corr_matrix[i,j]=cor_ij
    }
  }
  return(corr_matrix)
}


net_cor_matrix=cor_function(net_coverage)
rownames(net_cor_matrix)=c(&quot;NA18486&quot;, &quot;NA18505&quot;, &quot;NA18508&quot;, &quot;NA19128&quot;, &quot;NA19141&quot;, &quot;NA19193&quot;, &quot;NA19239&quot;, &#39;NA19257&#39;)
colnames(net_cor_matrix)=c(&quot;NA18486&quot;, &quot;NA18505&quot;, &quot;NA18508&quot;, &quot;NA19128&quot;, &quot;NA19141&quot;, &quot;NA19193&quot;, &quot;NA19239&quot;, &#39;NA19257&#39;)</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>Warning: package &#39;reshape2&#39; was built under R version 3.4.3</code></pre>
<pre class="r"><code>library(ggplot2)
melted_net_corr=melt(net_cor_matrix)
head(melted_net_corr)</code></pre>
<pre><code>     Var1    Var2     value
1 NA18486 NA18486 1.0000000
2 NA18505 NA18486 0.9555217
3 NA18508 NA18486 0.9526759
4 NA19128 NA18486 0.9799010
5 NA19141 NA18486 0.9822975
6 NA19193 NA18486 0.9699379</code></pre>
<pre class="r"><code>ggheatmap=ggplot(data = melted_net_corr, aes(x=Var1, y=Var2, fill=value)) +
  geom_tile() +
  labs(title=&quot;Net-seq Correlation Heatplot&quot;)
ggheatmap</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Look at the distribution of read counts:</p>
<ol style="list-style-type: decimal">
<li>Create a length of gene column to standardize by length.</li>
</ol>
<pre class="r"><code>summary(net_cov_full$`18486`)</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
      0.0       0.0       5.0     325.7      22.0 2651806.0 </code></pre>
<pre class="r"><code>plot(log(sort(net_cov_full$`18486`, decreasing = TRUE)))</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(log(sort(net_cov_full$`18505`, decreasing = TRUE)))</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>which(net_cov_full$`18486`== max(net_cov_full$`18486`))</code></pre>
<pre><code>[1] 12688 12689</code></pre>
<pre class="r"><code>net_cov_full[12688,]</code></pre>
<pre><code>      chr     start       end              gene score strand   18486
12688   2 118846049 118867604 ENST00000245787.4     0      + 2651806
         18505    18508    19128   19141    19193    19239    19257
12688 11856396 14059279 10862066 8216975 11877635 18071447 10156398</code></pre>
<pre class="r"><code>which(net_cov_full$`19239`== max(net_cov_full$`19239`))</code></pre>
<pre><code>[1] 12689</code></pre>
<pre class="r"><code>net_cov_full[12689,]</code></pre>
<pre><code>      chr     start       end              gene score strand   18486
12689   2 118846049 118868573 ENST00000485520.1     0      + 2651806
         18505    18508    19128   19141    19193    19239    19257
12689 11856396 14059279 10862067 8216975 11877635 18071449 10156398</code></pre>
<p>read buildup in insig2!</p>
<p>To look into this I will use bedtools genomecov. This will help me find the region where the build up is within this gene. I will do this for just one line first.</p>
<p>‘bedtools genomecov -ibam NA18152.bam -bg’</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=gencov_18486
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --output=gencov_sbatch.out
#SBATCH --error=gencov_sbatch.err
#SBATCH --mail-type=END

module load Anaconda3

source activate net-seq 


bedtools genomecov -ibam /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam -bg</code></pre>
<p>I made this file, now I can look for the region and find the base position of the buildup.</p>
<pre class="r"><code>gencov_18486=read.csv(&quot;../data/gencov_18486.bed&quot;, header=FALSE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)
colnames(gencov_18486)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;count&quot;)


gencov_18486_insig2= gencov_18486[gencov_18486$chr == &quot;2&quot; &amp; gencov_18486$start &gt;= 118846049 &amp; gencov_18486$end &lt;= 118868573, ]</code></pre>
<p>Buildup is 118847163-118867371</p>
<p><a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;lastVirtModeType=default&amp;lastVirtModeExtraState" class="uri">https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&amp;lastVirtModeType=default&amp;lastVirtModeExtraState</a>=&amp;virtModeType=default&amp;virtMode=0&amp;nonVirtPosition=&amp;position=chr2%3A118853899%2D118860634&amp;hgsid=657371987_p3YvC1JmckcMJLpbqv6SsBjfPeik</p>
<pre><code>http://genome.ucsc.edu/cgi-bin/das/hg19/dna?segment=chr2:118847163,118867371</code></pre>
<p>I will now be able to search for our primers in this seq to see if that is why we get buildup.<br />
GACGCTCTT</p>
<p>This sequence is in the reverse compliment of the linker:</p>
<p>/5rApp/(N)6AGATCGGAAGAGCGTC/3ddC</p>
<p>This is probably not long enough that this is the problem.</p>
<div id="mapping-statistics" class="section level3">
<h3>Mapping statistics:</h3>
<p>The quality score is in the sorted bam file.</p>
<p>MAPQ: MAPping Quality. It equals -10*log10 Prob mapping position is wrong, rounded to the nearest integer. A value 255 indicates that the mapping quality is not available.</p>
<p>I will get the mapping quality stats from the bam files as I did for Net1. I coppied the pytho script to /project2/gilad/briana/Net-seq/Net-seq3/code. I will create a wrap function for this in bash.</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=wrap_get_map
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

module load Anaconda3

source activate net-seq  

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18486-2018-1-8_S3_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_18486.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18505-2018-1-8_S1_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_18505.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-18508-2018-1-8_S2_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_18508.txt

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19128-2018-1-8_S8_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_19128.txt  

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19141-2018-1-8_S5_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_19141.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19193-2018-1-8_S6_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_19193.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19239-2018-1-8_S4_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_19239.txt 

python /project2/gilad/briana/Net-seq/Net-seq3/code/get_qual.py /project2/gilad/briana/Net-seq/Net-seq3/data/sort/YG-SP-NET3-19257-2018-1-8_S7_L008_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq3/output/mapped_qual_19257.txt 
</code></pre>
<pre class="r"><code>qual_mayer= read.csv(&quot;../data/qual_mayer.txt&quot;, head=FALSE)
summary(qual_mayer)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :13.00  
 Mean   :21.11  
 3rd Qu.:40.00  
 Max.   :40.00  </code></pre>
<p>I will compare our lines to the mayer data.</p>
<pre class="r"><code> mq_18486= read.csv(&quot;../data/mapped_qual_18486.txt&quot;, header=FALSE)
 mq_18505= read.csv(&quot;../data/mapped_qual_18505.txt&quot;, header=FALSE)
 mq_18508= read.csv(&quot;../data/mapped_qual_18508.txt&quot;, header=FALSE)
 mq_19128=read.csv(&quot;../data/mapped_qual_19128.txt&quot;, header=FALSE)
 mq_19141=read.csv(&quot;../data/mapped_qual_19141.txt&quot;, header=FALSE)
 mq_19193=read.csv(&quot;../data/mapped_qual_19193.txt&quot;, header=FALSE)
 mq_19239=read.csv(&quot;../data/mapped_qual_19239.txt&quot;, header=FALSE)
 mq_19257=read.csv(&quot;../data/mapped_qual_19257.txt&quot;, header=FALSE)</code></pre>
<p>summaries of each:</p>
<pre class="r"><code>summary(mq_18486)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :13.00  
 Mean   :17.46  
 3rd Qu.:20.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_18505)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :10.00  
 Mean   :14.41  
 3rd Qu.:13.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_18508)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :10.00  
 Mean   :14.11  
 3rd Qu.:13.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_19128)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :13.00  
 Mean   :18.75  
 3rd Qu.:20.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_19141)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :13.00  
 Mean   :17.39  
 3rd Qu.:20.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_19193)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:10.00  
 Median :13.00  
 Mean   :18.42  
 3rd Qu.:20.00  
 Max.   :40.00  </code></pre>
<pre class="r"><code>summary(mq_19239)</code></pre>
<pre><code>       V1      
 Min.   : 6.0  
 1st Qu.:10.0  
 Median :13.0  
 Mean   :16.3  
 3rd Qu.:20.0  
 Max.   :40.0  </code></pre>
<pre class="r"><code>summary(mq_19257)</code></pre>
<pre><code>       V1       
 Min.   : 6.00  
 1st Qu.:13.00  
 Median :13.00  
 Mean   :18.45  
 3rd Qu.:20.00  
 Max.   :40.00  </code></pre>
<p>Make histograms as I had for the net1.</p>
<pre class="r"><code>par(mfrow = c(3,3))

hist(as.numeric(mq_18486[,1]), freq=FALSE, main=&quot;Quality 18486&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_18505[,1]), freq=FALSE, main=&quot;Quality 18505&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_18508[,1]), freq=FALSE, main=&quot;Quality 18508&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_19128[,1]), freq=FALSE, main=&quot;Quality 19128&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_19141[,1]), freq=FALSE, main=&quot;Quality 19141&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_19193[,1]), freq=FALSE, main=&quot;Quality 19193&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_19239[,1]), freq=FALSE, main=&quot;Quality 19239&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(mq_19257[,1]), freq=FALSE, main=&quot;Quality 19257&quot;, xlab=&quot;Quality Score&quot;)
hist(as.numeric(qual_mayer[,1]), freq=FALSE, main= &quot;Quality Mayer&quot;, xlab=&quot;Quality Score&quot;)</code></pre>
<p><img src="figure/Net_3_explore.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_2.2.1  reshape2_1.4.3

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15     knitr_1.18       magrittr_1.5     munsell_0.4.3   
 [5] colorspace_1.3-2 rlang_0.1.6      stringr_1.2.0    plyr_1.8.4      
 [9] tools_3.4.2      grid_3.4.2       gtable_0.2.0     git2r_0.21.0    
[13] htmltools_0.3.6  yaml_2.1.16      lazyeval_0.2.1   rprojroot_1.3-2 
[17] digest_0.6.14    tibble_1.4.2     evaluate_0.10.1  rmarkdown_1.8.5 
[21] labeling_0.3     stringi_1.1.6    compiler_3.4.2   pillar_1.1.0    
[25] scales_0.5.0     backports_1.1.2 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
