<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2017-11-20" />

<title>Coverage in gene regions</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Coverage in gene regions</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2017-11-20</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-11-22</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 8465e92</p>
<div id="initial-exploration-for-data-format" class="section level2">
<h2>Initial exploration for data format</h2>
<p>The goal of this analysis is too look at read coverage in genes/exons/and in genes + promoter regions for my Net-seq data and the Mayer data. This will help me understand if the problem with this data is coverage.</p>
<p>I have downloaded 3 bed files from the UCSC table browser:</p>
<ul>
<li><p>Ref seq genes</p></li>
<li><p>Ref seq genes + 250 upstream</p></li>
<li><p>Ref seq exons</p></li>
</ul>
<p>I will use the bedtools coverage function. Documentation is below:</p>
<p><a href="http://bedtools.readthedocs.io/en/latest/content/tools/coverage.html" class="uri">http://bedtools.readthedocs.io/en/latest/content/tools/coverage.html</a></p>
<ul>
<li>a bed file for the genes/exons<br />
</li>
<li>bam files for my data and the mayer data</li>
</ul>
<pre class="bash"><code>
bedtools coverage -a &lt;FILE&gt; \
                             -b &lt;FILE1, FILE2, ..., FILEN&gt;</code></pre>
<p>Write a bash script for genes overlap: genome_overlap.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=bedtools_coverage
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --tasks-per-node=4 

#$1 reference to use 


bedtools coverage -a $1 -b /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18486-dep-2017-10-13_S4_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18508-dep-2017-10-13_S2_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18508-nondep-2017-10-13_S3_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-Unk1_S6_R1_001-sort.bam /project2/gilad/briana/Net-seq/data/sort/SRR1575922-sort.bam 

#&gt; to a txt file  


bedtools coverage -a ref_seq_gene_hg19_ -b /project2/gilad/briana/Net-seq/Net-seq1/data/net1_18486_dep_dedup_chr.bed -hist  &gt; gene_coverage_18486_dedup_hist.txt
</code></pre>
<p>Try with only 1 file first.</p>
<p>Try with /project2/gilad/briana/Net-seq/Net-seq1/data/net1_18486_dep_dedup_chr.bed</p>
<p>This works but it is for the deduplicated file. I am going to do the sorted files first. I need to convert the sorted bam files to bed files. All of the sorted bed files with the chr label are in data/bed</p>
<p>Should probably do this for the deduplicated files as well.</p>
<pre class="bash"><code>bedtools bamtobed [OPTIONS] -i &lt;BAM&gt;
awk &#39;$0=&quot;chr&quot;$0&#39; file &gt; new_file </code></pre>
<p>Only want first 6 columns of the ref file</p>
<pre class="bash"><code>

cat ref_seq_gene_hg19  | cut  -f 1-6 &gt; ref_seq_gene_hg19_small_cut
</code></pre>
<p>Still to big. Run on one file.</p>
<pre class="bash"><code>bedtools coverage -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed/net1_18486_dep_chr.bed  &gt; gene_coverage_18486_hist.txt</code></pre>
<p>Run this to make:</p>
<ul>
<li><p>gene_coverage_18508_dep_hist.txt</p></li>
<li><p>gene_coverage_18508_dep_hist.txt</p></li>
<li><p>gene_coverage_18508_nondep_hist.txt</p></li>
<li><p>gene_coverage_19238_dep_hist.txt</p></li>
<li><p>gene_coverage_mayer_SRR1575922_hist.txt</p></li>
</ul>
<p>Sort the bed files then rerun the coverage with the counts option to get one read count per gene. I have the script to sort the script files in each data/bed folder. ls</p>
<pre class="bash"><code>bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18486_dep_chr_sort.bed  &gt; gene_cov_count/gene_coverage_18486_count.txt


bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18508_dep_chr_sort.bed  &gt; gene_cov_count/gene_coverage_18508_dep_count.txt

bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18508_nondep_chr_sort.bed   &gt; gene_cov_count/gene_coverage_18508_nondep_count.txt

bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_19238_dep_chr_sort.bed   &gt; gene_cov_count/gene_coverage_19238_dep_count.txt


bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/data/bed_sort/mayer_SRR1575922_chr_sort.bed   &gt; gene_cov_count/gene_coverage_mayer_SRR1575922_count.txt
</code></pre>
</div>
<div id="data-input-non-depuplicated" class="section level2">
<h2>Data input (non-depuplicated)</h2>
<div id="gene-coverage" class="section level3">
<h3>Gene Coverage</h3>
<p>Install packages:</p>
<pre class="r"><code>library(vioplot)</code></pre>
<pre><code>Loading required package: sm</code></pre>
<pre><code>Package &#39;sm&#39;, version 2.2-5.4: type help(sm) for summary information</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<pre class="r"><code>gene_coverage_18486_count= read.csv(&quot;../data/gene_cov_count/gene_coverage_18486_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

gene_coverage_18508_dep_count= read.csv(&quot;../data/gene_cov_count/gene_coverage_18508_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

gene_coverage_18508_nondep_count= read.csv(&quot;../data/gene_cov_count/gene_coverage_18508_nondep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

gene_coverage_19238_dep_count= read.csv(&quot;../data/gene_cov_count/gene_coverage_19238_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

gene_coverage_mayer_dep_count = read.csv(&quot;../data/gene_cov_count/gene_coverage_mayer_SRR1575922_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>colnames(gene_coverage_18486_count) = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(gene_coverage_18508_dep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(gene_coverage_18508_nondep_count)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(gene_coverage_19238_dep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(gene_coverage_mayer_dep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)</code></pre>
<p>Next I will merge theses files to create 1 file with all of the data.</p>
<pre class="r"><code>gene_coverage_all= cbind(gene_coverage_18486_count, gene_coverage_18508_dep_count$counts, gene_coverage_18508_nondep_count$counts, gene_coverage_19238_dep_count$counts, gene_coverage_mayer_dep_count$counts)
colnames(gene_coverage_all)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;c_18486&quot;, &quot;c_18508_dep&quot;, &quot;c_18508_nondep&quot;, &quot;c_19238_dep&quot;, &quot;c_mayer&quot;)</code></pre>
<p>Add a column for length of A:</p>
<pre class="r"><code>#add length column  
gene_coverage_all= mutate(gene_coverage_all, length=end-start)
#add columns for standard count 
gene_coverage_all= mutate(gene_coverage_all,st_18486=c_18486/length)
gene_coverage_all= mutate(gene_coverage_all,st_18508_dep= c_18508_dep /length)
gene_coverage_all= mutate(gene_coverage_all,st_18508_nondep=c_18508_nondep/length)
gene_coverage_all= mutate(gene_coverage_all,st_19238=c_19238_dep/length)
gene_coverage_all= mutate(gene_coverage_all,st_mayer=c_mayer/length)</code></pre>
<pre class="r"><code>plot(sort(-log(gene_coverage_all$st_18486), decreasing=TRUE), ylab=&quot;-log of gene count standardized by length&quot;, xlab=&quot;Genes&quot;, main=&quot;18486&quot;)
abline(a=-log(mean(gene_coverage_all$st_18486)),b=0)</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(sort(-log(gene_coverage_all$st_18508_dep), decreasing=TRUE), ylab=&quot;-log of gene count standardized by length&quot;, xlab=&quot;Genes&quot;, main=&quot;18505 dep&quot;)
abline(a=-log(mean(gene_coverage_all$st_18508_dep)),b=0)</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(sort(-log(gene_coverage_all$st_18508_nondep), decreasing=TRUE), ylab=&quot;-log of gene count standardized by length&quot;, xlab=&quot;Genes&quot;, main=&quot;18505 nondep&quot;)
abline(a=-log(mean(gene_coverage_all$st_18508_nondep)),b=0)</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(sort(-log(gene_coverage_all$st_19238), decreasing=TRUE), ylab=&quot;-log of gene count standardized by length&quot;, xlab=&quot;Genes&quot;, main=&quot;19238&quot;)
abline(a=-log(mean(gene_coverage_all$st_19238)),b=0)</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(sort(-log(gene_coverage_all$st_mayer), decreasing=TRUE), ylab=&quot;-log of gene count standardized by length&quot;, xlab=&quot;Genes&quot;, main=&quot;Mayer&quot;)
abline(a=-log(mean(gene_coverage_all$st_mayer)),b=0)</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-12-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#boxplot- violin plot is better but you get infinities  
boxplot(-log(gene_coverage_all$st_18486), -log(gene_coverage_all$st_18508_dep), -log(gene_coverage_all$st_18508_nondep), -log(gene_coverage_all$st_19238), -log(gene_coverage_all$st_mayer), names=c(&quot;18486&quot;, &quot;18508_dep&quot;, &quot;18508_nondep&quot;, &quot;19238&quot;, &quot;Mayer&quot;), ylab=&quot;-log of standard expression&quot;)</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (Inf) in boxplot 1 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (Inf) in boxplot 2 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (Inf) in boxplot 3 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (Inf) in boxplot 4 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (Inf) in boxplot 5 is not drawn</code></pre>
<p><img src="figure/reads_in_genes.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>To use ggplot I need to reformat the dataframe to gene by sample.</p>
<pre class="r"><code>gene_names= gene_coverage_all$name
standard_counts= gene_coverage_all[, 13:17]
#row.names(standard_counts)=gene_names</code></pre>
</div>
<div id="gene-and-promoter-coverage" class="section level3">
<h3>Gene and promoter coverage</h3>
<pre class="r"><code>prom_coverage_18486_count= read.csv(&quot;../data/prom_coverage/prom_coverage_18486_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

prom_coverage_18508_dep_count= read.csv(&quot;../data/prom_coverage/prom_coverage_18508_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

prom_coverage_18508_nondep_count= read.csv(&quot;../data/prom_coverage/prom_coverage_18508_nondep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

prom_coverage_19238_dep_count= read.csv(&quot;../data/prom_coverage/prom_coverage_19238_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

prom_coverage_mayer_dep_count = read.csv(&quot;../data/prom_coverage/promoter_coverage_mayer_SRR1575922_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>colnames(prom_coverage_mayer_dep_count)=  c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(prom_coverage_19238_dep_count)=  c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(prom_coverage_18508_nondep_count)=  c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(prom_coverage_18508_dep_count) =  c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)  
colnames(prom_coverage_18486_count)=  c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)</code></pre>
</div>
<div id="exon-coverage" class="section level3">
<h3>Exon coverage</h3>
<pre class="r"><code>exon_coverage_18486_count= read.csv(&quot;../data/exon_cov/exon_coverage_18486_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

exon_coverage_18508_dep_count= read.csv(&quot;../data/exon_cov/exon_coverage_18508_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

exon_coverage_18508_nondep_count= read.csv(&quot;../data/exon_cov/exon_coverage_18508_nondep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

exon_coverage_19238_dep_count= read.csv(&quot;../data/exon_cov/exon_coverage_19238_dep_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)

exon_coverage_mayer_dep_count = read.csv(&quot;../data/exon_cov/exon_coverage_mayer_SRR1575922_count.txt&quot;, header=FALSE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>colnames(exon_coverage_mayer_dep_count)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(exon_coverage_19238_dep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(exon_coverage_18508_nondep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(exon_coverage_18508_dep_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)
colnames(exon_coverage_18486_count)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;counts&quot;)  </code></pre>
</div>
<div id="coverage-with-histograms" class="section level3">
<h3>Coverage with histograms</h3>
<p>These are the historgram files I made.</p>
<pre class="r"><code>#gene_coverage_18486_dedup= read.csv(&quot;../data/gene_coverage_18486_dedup_hist.txt&quot;, header=FALSE, sep=&quot;\t&quot;)


#gene_coverage_18486= read.csv(&quot;../data/gene_coverage_18486_hist.txt&quot;, head=FALSE, sep=&quot;\t&quot;)

#gene_coverage_18508_dep= read.csv(&quot;../data/gene_coverage_18508_dep_hist.txt&quot;, header=FALSE, sep=&quot;\t&quot;)
#gene_coverage_18508_nondep= read.csv(&quot;../data/gene_coverage_18508_nondep_hist.txt&quot;, head=FALSE, sep=&quot;\t&quot;)
#gene_coverage_19238_dep = read.csv(&quot;../data/gene_coverage_19238_dep_hist.txt&quot;, head=FALSE, sep=&quot;\t&quot;)
#gene_coverage_mayer_dep = read.csv(&quot;../data/gene_coverage_mayer_SRR1575922_hist.txt&quot;, head=FALSE, sep=&quot;\t&quot;)</code></pre>
<p>Add column names:</p>
<pre class="r"><code>#colnames(gene_coverage_18486_dedup)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)
#colnames(gene_coverage_18486)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)
#colnames(gene_coverage_18508_dep)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)
#colnames(gene_coverage_18508_nondep)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)
#colnames(gene_coverage_19238_dep)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)
#colnames(gene_coverage_mayer_dep)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;overlap&quot;, &quot;bases_non_zero&quot;, &quot;lengthA&quot;, &quot;frac_A_non_zero_hist&quot;)</code></pre>
<p>Omit NA columns:</p>
<pre class="r"><code># gene_coverage_18486= na.omit(gene_coverage_18486)
# gene_coverage_18486_dedup= na.omit(gene_coverage_18486_dedup)
# gene_coverage_18508_dep= na.omit(gene_coverage_18508_hist_dep)
# gene_coverage_18508_nondep= na.omit(gene_coverage_18508_nondep)
# gene_coverage_19238_dep= na.omit(gene_coverage_19238_dep)
# gene_coverage_mayer_dep= na.omit(gene_coverage_mayer_dep)</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2  ggplot2_2.2.1 dplyr_0.7.4   vioplot_0.2   sm_2.2-5.4   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13     knitr_1.17       bindr_0.1        magrittr_1.5    
 [5] munsell_0.4.3    colorspace_1.3-2 R6_2.2.2         rlang_0.1.4     
 [9] plyr_1.8.4       stringr_1.2.0    tools_3.4.2      grid_3.4.2      
[13] gtable_0.2.0     git2r_0.19.0     htmltools_0.3.6  lazyeval_0.2.1  
[17] yaml_2.1.14      rprojroot_1.2    digest_0.6.12    assertthat_0.2.0
[21] tibble_1.3.4     glue_1.2.0       evaluate_0.10.1  rmarkdown_1.6   
[25] stringi_1.1.5    compiler_3.4.2   scales_0.5.0     backports_1.1.1 
[29] pkgconfig_2.0.1 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
