<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-03-22" />

<title>Strand Specificity of Net seq data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Strand Specificity of Net seq data</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2018-03-22</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-03-26</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 8e2043e</p>
<!-- Add your analysis here -->
<p>The purpose of this analysis is to explore the strand specificity of my net seq data. Within genes we expect strand specific expression and at TSS we would expect some convergent transcription. I will filter the overlaping genes as done in the Meyer paper.</p>
<p>Mayer filtering: Pol-II-transcribed genes that do not overlap other genes within 2.5 kb of the TSS and polyadenylation site and are longer than 2 kb</p>
<div id="filter-the-genes" class="section level3">
<h3>Filter the genes</h3>
<p>bedtools closest -N</p>
<p>Get the protein coding genes:</p>
<pre class="bash"><code>grep &#39;protein_coding&#39; gencode.v19.annotation.bed  | awk &#39;$8 == &quot;gene&quot;&#39; |cut -f1-6  |sed &#39;s/^chr//&#39; &gt;  gencode.v19.annotation.proteincodinggene.bed </code></pre>
<p>Only keep genes that are 2kb. I want genes where end (3) - start (2) is greater than 2000</p>
<pre class="bash"><code>awk &#39;{if( ($3 - $2) &gt; 2000)  print($1 &quot;\t&quot; $2 &quot;\t&quot; $3 &quot;\t&quot; $4 &quot;\t&quot; $5 &quot;\t&quot; $6)}&#39; gencode.v19.annotation.proteincodinggene.bed &gt; gencode.v19.annotation.proteincodinggene.2kb.bed</code></pre>
<p>Next step is to look for overlab distance with bedtools closest -N.</p>
<p>gene_dist.sh</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=gene_dist
#SBATCH --time=8:00:00
#SBATCH --output=gene_dist.out
#SBATCH --error=gene_dist.err
#SBATCH --partition=gilad
#SBATCH --mem=10G
#SBATCH --mail-type=END

module load Anaconda3  

source activate net-seq 

bedtools closest -N -d -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.proteincodinggene.2kb.bed -b /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.proteincodinggene.2kb.bed &gt; /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.proteincodinggene.2kb.dist.bed</code></pre>
<p>Column 13 of this file must be &gt; 2500</p>
<pre class="bash"><code>awk &#39;{if( $13 &gt; 2500)  print($1 &quot;\t&quot; $2 &quot;\t&quot; $3 &quot;\t&quot; $4 &quot;\t&quot; $5 &quot;\t&quot; $6)}&#39; gencode.v19.annotation.proteincodinggene.2kb.dist.bed &gt; gencode.v19.annotation.distfilteredgenes.bed</code></pre>
<p>This leaves 9367 genes.</p>
</div>
<div id="get-strand-specific-gene-counts-for-these-genes" class="section level3">
<h3>Get strand specific gene counts for these genes</h3>
<p>Do this for 18486 first.</p>
<p>Sort both files with sort -k1,1 -k2,2n</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=strand
#SBATCH --time=8:00:00
#SBATCH --output=strand.out
#SBATCH --error=strnad.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END



bedtools coverage -counts -s -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18486_combined_Netpilot-bedsort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/same_strand_genecounts_18486.txt 


bedtools coverage -counts -S -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18486_combined_Netpilot-bedsort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/opp_strand_genecounts_18486.txt </code></pre>
</div>
<div id="process-data" class="section level3">
<h3>Process data</h3>
<p>Library input:</p>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>Loading required package: rmarkdown</code></pre>
<pre><code>This is workflowr version 0.7.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
<p>Input data:</p>
<pre class="r"><code>opp_strand=read.table(&quot;../data/same_strand_genecounts_18486.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(opp_strand)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;opp_count&quot;)
same_strand=read.table(&quot;../data/opp_strand_genecounts_18486.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(same_strand)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;same_count&quot;)</code></pre>
<p>Merge on the gene names:</p>
<pre class="r"><code>full_strand=inner_join(same_strand,opp_strand, by=c(c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;)))
full_strand= full_strand %&gt;% mutate(perc_same=(same_count /(same_count + opp_count))) %&gt;% filter(perc_same != &quot;NaN&quot;)</code></pre>
</div>
<div id="explore-data" class="section level3">
<h3>Explore data:</h3>
<pre class="r"><code>summary(full_strand$perc_same)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.5000  0.8333  0.6927  0.9611  1.0000 </code></pre>
<pre class="r"><code>plot(sort(full_strand$perc_same, decreasing=TRUE), col=ifelse((sort(full_strand$perc_same, decreasing=TRUE)) &gt;.5, &quot;red&quot;, &quot;black&quot;), ylab=&quot;same strand / gene count&quot;, xlab=&quot;Genes ordered by strand spec percent&quot;, main=&quot;Gene counts strand specificity 18486&quot;)
legend(&quot;topright&quot;, legend = c(&quot;top 75%&quot;), col=c(&quot;red&quot;), pch=16, cex=1)</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log10(full_strand$same_count), log10(full_strand$opp_count), names=c(&quot;Same strand&quot;, &quot;Opposite Strand&quot;), ylab=&quot;log10 Gene counts&quot;, main=&quot;Gene counts by strand&quot;)</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 1 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 2 is not drawn</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I am going to run the gene counts analysis not strand specific to make sure I am getting a sum of the these counts.</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=all_counts
#SBATCH --time=8:00:00
#SBATCH --output=all_strand.out
#SBATCH --error=all_strnad.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END



bedtools coverage -counts  -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18486_combined_Netpilot-bedsort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/all_strand_genecounts_18486.txt 
</code></pre>
<p>This file confirmed that the counts without specificity is the sum of the strand information.</p>
</div>
<div id="try-this-analysis-in-another-line" class="section level3">
<h3>Try this analysis in another line</h3>
<p>strand_spec_18505.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=strand_18505
#SBATCH --time=8:00:00
#SBATCH --output=strand.out
#SBATCH --error=strnad.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END


bedtools coverage -counts -s -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18505_combined_Netpilot-bedsort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/same_strand_genecounts_18505.txt 


bedtools coverage -counts -S -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18505_combined_Netpilot-bedsort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/opp_strand_genecounts_18505.txt </code></pre>
<pre class="r"><code>opp_strand_18505=read.table(&quot;../data/same_strand_genecounts_18505.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(opp_strand_18505)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;opp_count&quot;)
same_strand_18505=read.table(&quot;../data/opp_strand_genecounts_18505.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(same_strand_18505)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;same_count&quot;)


full_strand_18505=inner_join(same_strand_18505,opp_strand_18505, by=c(c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;)))
full_strand_18505= full_strand_18505 %&gt;% mutate(perc_same=(same_count /(same_count + opp_count))) %&gt;% filter(perc_same != &quot;NaN&quot;)

summary(full_strand_18505$perc_same)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.4545  0.8333  0.6850  0.9722  1.0000 </code></pre>
<pre class="r"><code>plot(sort(full_strand_18505$perc_same, decreasing = TRUE), col=ifelse((sort(full_strand_18505$perc_same, decreasing = TRUE))&gt;.45, &quot;red&quot;, &quot;black&quot;), ylab=&quot;same strand / gene count&quot;, xlab=&quot;Genes ordered by strand spec percent&quot;, main=&quot;Gene counts strand specificity 18505&quot;)
legend(&quot;topright&quot;, legend = c(&quot;top 75%&quot;), col=c(&quot;red&quot;), pch=16, cex=1)</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log10(full_strand_18505$same_count), log10(full_strand_18505$opp_count), names=c(&quot;Same strand&quot;, &quot;Opposite Strand&quot;), ylab=&quot;log10 Gene counts&quot;, main=&quot;Gene counts by strand 18505&quot;)</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 1 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 2 is not drawn</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distribution-for-genes-1-18486" class="section level3">
<h3>Distribution for genes &gt;1 18486</h3>
<pre class="r"><code>gene_counts_18486=read.table(&quot;../data/all_strand_genecounts_18486.txt&quot;, stringsAsFactors = FALSE, header=FALSE)
colnames(gene_counts_18486)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;count&quot;)
mapped_18486_mil= 65189389 / 10^6
gene_counts_18486=gene_counts_18486 %&gt;% mutate(K_length= (end - start)/100) %&gt;% mutate(rpkm= + count/(K_length * mapped_18486_mil)) %&gt;% filter(rpkm&gt;=1)</code></pre>
<p>Join this with the strand specific information.</p>
<pre class="r"><code>top_genes_strand= inner_join(full_strand, gene_counts_18486, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;))</code></pre>
<p>Plot this distribution:</p>
<pre class="r"><code>plot(sort(top_genes_strand$perc_same, decreasing = TRUE), ylab=&quot;same strand / gene count&quot;, xlab=&quot;Genes ordered by strand spec percent&quot;, main=&quot;Gene counts strand specificity 18486 for genes &gt;1 RPKM&quot;)</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Box plots:</p>
<pre class="r"><code>boxplot(log10(top_genes_strand$same_count), log10(top_genes_strand$opp_count), names=c(&quot;Same strand&quot;, &quot;Opposite Strand&quot;), ylab=&quot;log10 Gene counts&quot;, main=&quot;Gene counts by strand, genes &gt;1 RPKM&quot;)</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 1 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 2 is not drawn</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-out-the-small-rnas-that-fall-in-these-genes" class="section level3">
<h3>Filter out the small RNAs that fall in these genes</h3>
<p>Filter the bams as I did in the create blacklist. filt_smRNA_bam.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=intersect_bam
#SBATCH --time=8:00:00
#SBATCH --output=intbam_sbatch.out
#SBATCH --error=intbam_sbatch.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END


module load Anaconda3  

source activate net-seq 


sample=$1

describer=$(echo ${sample} | sed -e &#39;s/.*\YG-SP-//&#39; | sed -e &quot;s/_combined_Netpilot-sort.bam$//&quot;)


bedtools intersect -v -wa -abam $1 -b /project2/gilad/briana/genome_anotation_data/snRNA.gencode.v19.nochr.bed /project2/gilad/briana/genome_anotation_data/snoRNA.gencode.v19.nochr.bed  &gt; /project2/gilad/briana/Net-seq-pilot/data/filter_bam/${describer}.rnafilter.bam
</code></pre>
<p>run on: /project2/gilad/briana/Net-seq-pilot/data/sort/YG-SP-NET3-18486_combined_Netpilot-sort.bam</p>
<p>Now when I look at the bams in IGV I have removed the snoRNA in TPT1.</p>
</div>
<div id="rerun-analysis-with-filtered-bam" class="section level3">
<h3>Rerun analysis with Filtered bam</h3>
<p>This time I will switch the -s and -S so we get the correct same/opp strand information in the txt files.</p>
<ol style="list-style-type: decimal">
<li>Convert the filtered bam file to a bed file and sort it:</li>
</ol>
<pre class="bash"><code>bamToBed -i NET3-18486.rnafilter.bam &gt; ../bed/YG-SP-NET3-18486_combined_Netpilot-rnafilt.bed


sort -k1,1 -k2,2n YG-SP-NET3-18486_combined_Netpilot-rnafilt.bed &gt; YG-SP-NET3-18486_combined_Netpilot-rnafilt-sort.bed</code></pre>
<p>strand_filt.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=_filtstrand
#SBATCH --time=8:00:00
#SBATCH --output=filtstrand.out
#SBATCH --error=filtstrnad.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END



bedtools coverage -counts -S -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18486_combined_Netpilot-rnafilt-sort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/same_strand_genecounts_filt_18486.txt 


bedtools coverage -counts -s -sorted -a /project2/gilad/briana/genome_anotation_data/gencode.v19.annotation.distfilteredgenes.sort.bed -b /project2/gilad/briana/Net-seq-pilot/data/bed/YG-SP-NET3-18486_combined_Netpilot-rnafilt-sort.bed  &gt; /project2/gilad/briana/Net-seq-pilot/output/strand_spec/opp_strand_genecounts_filt_18486.txt </code></pre>
<pre class="r"><code>opp_strand_18486_filt=read.table(&quot;../data/opp_strand_genecounts_filt_18486.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(opp_strand_18486_filt)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;opp_count&quot;)
same_strand_18486_filt=read.table(&quot;../data/same_strand_genecounts_filt_18486.txt&quot;, header=FALSE, stringsAsFactors = FALSE)
colnames(same_strand_18486_filt)= c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;, &quot;same_count&quot;)


full_strand_18486_filt=inner_join(same_strand_18486_filt, opp_strand_18486_filt, by=c(c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;,&quot;score&quot;, &quot;strand&quot;)))
full_strand_18486_filt= full_strand_18486_filt %&gt;% mutate(perc_same=(same_count /(same_count + opp_count))) %&gt;% filter(perc_same != &quot;NaN&quot;)

summary(full_strand_18486_filt$perc_same)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.5000  0.8319  0.6927  0.9594  1.0000 </code></pre>
<pre class="r"><code>plot(sort(full_strand_18486_filt$perc_same, decreasing = TRUE), col=ifelse((sort(full_strand_18486_filt$perc_same, decreasing = TRUE))&gt;.5, &quot;red&quot;, &quot;black&quot;), ylab=&quot;same strand / gene count&quot;, xlab=&quot;Genes ordered by strand spec percent&quot;, main=&quot;Gene counts strand specificity filtered 18486&quot;)
legend(&quot;topright&quot;, legend = c(&quot;top 75%&quot;), col=c(&quot;red&quot;), pch=16, cex=1)</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log10(full_strand_18486_filt$same_count), log10(full_strand_18486_filt$opp_count), names=c(&quot;Same strand&quot;, &quot;Opposite Strand&quot;), ylab=&quot;log10 Gene counts&quot;, main=&quot;Gene counts by strand 18486 filtered&quot;)</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 1 is not drawn</code></pre>
<pre><code>Warning in bplt(at[i], wid = width[i], stats = z$stats[, i], out = z$out[z
$group == : Outlier (-Inf) in boxplot 2 is not drawn</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code> hist(full_strand_18486_filt$perc_same, xlab=&quot;Same strand/full count for gene&quot;, main=&quot;18486 strand specificity on filtered genes and filtered bams&quot;)</code></pre>
<p><img src="figure/strand_spec.Rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2    ggplot2_2.2.1   workflowr_0.7.0 rmarkdown_1.8.5
[5] dplyr_0.7.4    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15     knitr_1.18       bindr_0.1        magrittr_1.5    
 [5] munsell_0.4.3    colorspace_1.3-2 R6_2.2.2         rlang_0.1.6     
 [9] plyr_1.8.4       stringr_1.2.0    tools_3.4.2      grid_3.4.2      
[13] gtable_0.2.0     git2r_0.21.0     htmltools_0.3.6  lazyeval_0.2.1  
[17] yaml_2.1.16      rprojroot_1.3-2  digest_0.6.14    assertthat_0.2.0
[21] tibble_1.4.2     glue_1.2.0       evaluate_0.10.1  stringi_1.1.6   
[25] compiler_3.4.2   pillar_1.1.0     scales_0.5.0     backports_1.1.2 
[29] pkgconfig_2.0.1 </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
