<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-01-24" />

<title>3` UTR analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Net-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/Net-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">3` UTR analysis</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2018-01-24</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-01-26</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4aeaf94</p>
<!-- Add your analysis here -->
<p>I have a file with the 3` UTRs and I want to extract coverage from the drop and dronc seq bam files that are in these UTR regions.</p>
<p>I will use:</p>
<ul>
<li>bedtools coverage -d -a /project2/gilad/spott/dropBams/Day7_cardiomyocytes_drop_seq.bam -b /project2/gilad/briana/apa_sites/three_prime_utr.bed &gt; drop7_cardio_3utr.txt</li>
</ul>
<p>This is taking too much memory. I am going to convert bam to a bed file then sort it.</p>
<pre class="bash"><code>bedtools bamtobed -i /project2/gilad/spott/dropBams/Day7_cardiomyocytes_drop_seq.bam &gt; Day7_cardiomyocytes_drop_seq.bed  

sort -k1,1 -k2,2n Day7_cardiomyocytes_drop_seq.bed &gt;  Day7_cardiomyocytes_drop_seq_sort.bed </code></pre>
<p>Do this for the UTR file as well.</p>
<pre class="bash"><code>sort -k1,1 -k2,2n three_prime_utr.bed &gt; three_prime_utr_sort.bed  </code></pre>
<p>Now add the sorted to the coverage: Update the drop7_cardio_3UTR_cov.sh script</p>
<pre class="bash"><code>bedtools coverage -d -sorted  -a /project2/gilad/briana/apa_sites/Day7_cardiomyocytes_drop_seq_sort.bed  -b /project2/gilad/briana/apa_sites/three_prime_utr_sort.bed &gt; drop7_cardio_3utr.txt</code></pre>
<p>Do this for one of 3â€™ seq lines too (LCL). The scipt is called lcl_3UTR_cov.sh.</p>
<p><em>These files are too big to move onto my computer. I need to think of anotehr way to vizualize them.</em></p>
<div id="called-apa-analysis" class="section level3">
<h3>Called ApA analysis</h3>
<p>Alternative anaylsis: Try to look at this similar to how I looked at the TSS enrichment.</p>
<ul>
<li><p>Step 1: Download drop and dronc seq bam/ index files to my computer in the netseq data file.</p></li>
<li><p>clusters.hg38</p></li>
<li><p>Day7_cardiomyocytes_drop_seq.bam</p></li>
<li><p>Day7_cardiomyocytes_drop_seq.bam.bai</p></li>
<li><p>three_prime_utr.bed</p></li>
<li><p>Step 2: Pull in packages and data for analysis:</p></li>
</ul>
<pre class="r"><code>#get reads
reads &lt;- readGAlignments(file = &quot;../data/Day7_cardiomyocytes_drop_seq.bam&quot;, index=&quot;../data/Day7_cardiomyocytes_drop_seq.bam.bai&quot;)
reads.GR &lt;-  granges(reads)

UTR=readGeneric(&quot;../data/three_prime_utr.bed&quot;)
pAsite= readGeneric(&quot;../data/clusters.hg38.bed&quot;)</code></pre>
<pre class="r"><code>#resize so I am looking 10,000 up and downstream of the center of the UTR
UTR %&lt;&gt;% resize(., width=10000, fix=&quot;center&quot;)
(UTR_width= summary(width(UTR)))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  10000   10000   10000   10000   10000   10000 </code></pre>
<pre class="r"><code>sm &lt;- ScoreMatrixBin(target = reads.GR,  windows = UTR, bin.num = 100, bin.op = &quot;mean&quot;)  </code></pre>
<pre><code>Warning in .local(target, windows, bin.num, bin.op, strand.aware): 5723
windows fall off the target</code></pre>
<pre class="r"><code>plotMeta(sm)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Do this against the pAsites:</p>
<pre class="r"><code>(pAs_width= summary(width(pAsite)))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2.00    8.00   13.00   13.13   18.00   61.00 </code></pre>
<pre class="r"><code>#look 200 up and down stream of each 
pAsite %&lt;&gt;% resize(., width=500, fix=&quot;center&quot;)
(pAs_width2= summary(width(pAsite)))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    500     500     500     500     500     500 </code></pre>
<pre class="r"><code>sm_pA &lt;- ScoreMatrixBin(target = reads.GR,  windows = pAsite, bin.num = 500, bin.op = &quot;mean&quot;,strand.aware=TRUE)</code></pre>
<pre><code>Warning in .local(target, windows, bin.num, bin.op, strand.aware): 10
windows fall off the target</code></pre>
<pre class="r"><code>plotMeta(sm_pA, xcoords = c(-250,250), main=&quot;pA enrichment Day7 Drop&quot;)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /> Try with the Dronc seq data:</p>
<pre class="r"><code>dronc_reads &lt;- readGAlignments(file = &quot;../data/Day7_cardiomyocytes_droNC_seq.bam&quot;, index=&quot;../data/Day7_cardiomyocytes_droNC_seq.bam.bai&quot;)
dronc_reads.GR &lt;-  granges(dronc_reads)


dronc_sm_pA &lt;- ScoreMatrixBin(target =dronc_reads.GR,  windows = pAsite, bin.num = 500, bin.op = &quot;mean&quot;, strand.aware=TRUE)</code></pre>
<pre><code>Warning in .local(target, windows, bin.num, bin.op, strand.aware): 10
windows fall off the target</code></pre>
<pre class="r"><code>plotMeta(dronc_sm_pA, xcoords = c(-250,250),  main=&quot;pA enrichment Day7 Dronc&quot;)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Compare this result to the 3` seq data :</p>
<ul>
<li>LCL</li>
</ul>
<pre class="r"><code>LCL_reads &lt;- readGAlignments(file = &quot;../data/blcl.hg38.sorted.bam&quot;, index=&quot;../data/blcl.hg38.sorted.bam.bai&quot;)
LCL_reads.GR &lt;-  granges(LCL_reads)

sm_LCL_pA &lt;- ScoreMatrixBin(target = LCL_reads.GR,  windows = pAsite, bin.num = 500, bin.op = &quot;mean&quot;, strand.aware=TRUE)
plotMeta(sm_LCL_pA, xcoords = c(-250,250),  main=&quot;pA enrichment LCL&quot;)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>hES</li>
</ul>
<pre class="r"><code>hES_reads &lt;- readGAlignments(file = &quot;../data/hES.hg38.sorted.bam&quot;, index=&quot;../data/hES.hg38.sorted.bam.bai&quot;) 
hES_reads.GR &lt;-  granges(hES_reads)

sm_hES_pA &lt;- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite, bin.num = 500, bin.op = &quot;mean&quot;, strand.aware=TRUE) 
plotMeta(sm_hES_pA, xcoords = c(-250,250),  main=&quot;pA enrichment hES&quot;) </code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I now want to try to look at the 3` most pAs. Start by overlapping the PaS sites with the UTRs then annotate the file with the UTR name.</p>
<p>bedtools intersect- I want only the As (pAs) that are in B (UTR)</p>
<ul>
<li>bedtools intersect -a clusters.hg38.bed -b three_prime_utr.bed &gt; clusters.hg38.3utr.bed</li>
</ul>
<p>Alternative methed: subset the pAS in the 3â€™ UTRs then seperate the files by strandedness.</p>
<pre class="r"><code>pAsite_pos= readGeneric(&quot;../data/clusters.hg38.3utr.pos.bed&quot;)
pAsite_pos %&lt;&gt;% resize(., width=1000, fix=&quot;center&quot;)
(pAs_pos_width= summary(width(pAsite_pos)))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1000    1000    1000    1000    1000    1000 </code></pre>
<pre class="r"><code>pAsite_neg= readGeneric(&quot;../data/clusters.hg38.3utr.neg.bed&quot;)
pAsite_neg %&lt;&gt;% resize(., width=1000, fix=&quot;center&quot;)</code></pre>
<pre class="r"><code>#drop and pos strand
sm_pA_pos &lt;- ScoreMatrixBin(target = reads.GR,  windows = pAsite_pos, bin.num = 1000, bin.op = &quot;mean&quot;)
plotMeta(sm_pA_pos, xcoords = c(-500,500), main=&quot;pA enrichment Day7 Drop pos strand&quot;)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /> negative Graph:</p>
<pre class="r"><code>#drop and neg strand
sm_pA_neg &lt;- ScoreMatrixBin(target = reads.GR,  windows = pAsite_neg, bin.num = 1000, bin.op = &quot;mean&quot;)
plotMeta(sm_pA_neg, xcoords = c(-500,500), main=&quot;pA enrichment Day7 Drop neg strand&quot;)</code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>hES with pos</p>
<pre class="r"><code>sm_hES_pA_pos &lt;- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite_pos, bin.num = 1000, bin.op = &quot;mean&quot;) 
plotMeta(sm_hES_pA_pos, xcoords = c(-500,500),  main=&quot;pA enrichment hES pos&quot;) </code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>hES neg</p>
<pre class="r"><code>sm_hES_pA_neg &lt;- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite_neg, bin.num = 1000, bin.op = &quot;mean&quot;) 
plotMeta(sm_hES_pA_neg, xcoords = c(-500,500),  main=&quot;pA enrichment hES pos&quot;) </code></pre>
<p><img src="figure/three_prime_UTR.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heat-maps-with-enrichedheatmap" class="section level3">
<h3>Heat maps with EnrichedHeatmap</h3>
<pre class="r"><code>library(EnrichedHeatmap)</code></pre>
<pre><code>Loading required package: ComplexHeatmap</code></pre>
<pre><code>========================================
ComplexHeatmap version 1.17.1
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://bioconductor.org/packages/ComplexHeatmap/

If you use it in published research, please cite:
Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
  genomic data. Bioinformatics 2016.
========================================</code></pre>
<pre><code>Loading required package: locfit</code></pre>
<pre><code>locfit 1.5-9.1   2013-03-22</code></pre>
<pre><code>========================================
EnrichedHeatmap version 1.9.2
Bioconductor page: http://bioconductor.org/packages/EnrichedHeatmap/
Github page: https://github.com/jokergoo/EnrichedHeatmap
Documentation: http://bioconductor.org/packages/EnrichedHeatmap/
========================================</code></pre>
<pre><code>
Attaching package: &#39;EnrichedHeatmap&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ComplexHeatmap&#39;:

    +.AdditiveUnit</code></pre>
<p>Drop seq day7 cardiomyocytes GRanges object:</p>
<pre class="r"><code>reads.GR[1:5]</code></pre>
<pre><code>GRanges object with 5 ranges and 0 metadata columns:
      seqnames         ranges strand
         &lt;Rle&gt;      &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1 [15855, 15916]      -
  [2]     chr1 [16449, 16510]      -
  [3]     chr1 [16449, 16510]      -
  [4]     chr1 [16449, 16510]      -
  [5]     chr1 [16449, 16510]      -
  -------
  seqinfo: 194 sequences from an unspecified genome</code></pre>
<p>pAs:</p>
<pre class="r"><code>pAsite[1:5]</code></pre>
<pre><code>GRanges object with 5 ranges and 0 metadata columns:
      seqnames           ranges strand
         &lt;Rle&gt;        &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1 [628991, 629490]      *
  [2]     chr1 [629005, 629504]      *
  [3]     chr1 [629025, 629524]      *
  [4]     chr1 [629042, 629541]      *
  [5]     chr1 [629056, 629555]      *
  -------
  seqinfo: 43 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>#mat_drop7_pAs = normalizeToMatrix(UTR , pAsite, value_column = &quot;ranges&quot;, extend = 1000, mean_mode = &quot;w0&quot;, w = 100)</code></pre>
<pre class="r"><code># library(biomaRt)
# mart = useMart(biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
#                dataset = &quot;hsapiens_gene_ensembl&quot;)
# filterlist &lt;- c(1:22, &#39;X&#39;, &#39;Y&#39;)
# ds = useDataset(&#39;hsapiens_gene_ensembl&#39;, mart = mart)
# 
# egs = getBM(attributes = c(&#39;ensembl_gene_id&#39;, &#39;external_gene_name&#39;, &#39;chromosome_name&#39;, &#39;start_position&#39;, &#39;end_position&#39;, &#39;strand&#39;), 
#             filters = &#39;chromosome_name&#39;,
#             values = filterlist,
#             mart = ds)
# reads.GR_chr &lt;-  granges(reads, seqnames = Rle(paste0(&#39;chr&#39;, egs$chromosome_name )))
# pAsite_chr= readGeneric(&quot;../data/clusters.hg38.bed&quot;, seqnames = Rle(paste0(&#39;chr&#39;, egs$chromosome_name )))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] EnrichedHeatmap_1.9.2      locfit_1.5-9.1            
 [3] ComplexHeatmap_1.17.1      GenomicAlignments_1.14.1  
 [5] Rsamtools_1.30.0           Biostrings_2.46.0         
 [7] XVector_0.18.0             SummarizedExperiment_1.8.1
 [9] DelayedArray_0.4.1         matrixStats_0.53.0        
[11] Biobase_2.38.0             BiocInstaller_1.28.0      
[13] magrittr_1.5               data.table_1.10.4-3       
[15] genomation_1.10.0          dplyr_0.7.4               
[17] GenomicRanges_1.30.1       GenomeInfoDb_1.14.0       
[19] IRanges_2.12.0             S4Vectors_0.16.0          
[21] BiocGenerics_0.24.0       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15           circlize_0.4.3         lattice_0.20-35       
 [4] assertthat_0.2.0       rprojroot_1.3-2        digest_0.6.14         
 [7] gridBase_0.4-7         R6_2.2.2               plyr_1.8.4            
[10] backports_1.1.2        evaluate_0.10.1        ggplot2_2.2.1         
[13] pillar_1.1.0           GlobalOptions_0.0.12   zlibbioc_1.24.0       
[16] rlang_0.1.6            lazyeval_0.2.1         GetoptLong_0.1.6      
[19] Matrix_1.2-12          rmarkdown_1.8.5        BiocParallel_1.12.0   
[22] readr_1.1.1            stringr_1.2.0          RCurl_1.95-4.10       
[25] munsell_0.4.3          compiler_3.4.2         rtracklayer_1.38.3    
[28] pkgconfig_2.0.1        shape_1.4.3            htmltools_0.3.6       
[31] tibble_1.4.2           GenomeInfoDbData_1.0.0 XML_3.98-1.9          
[34] bitops_1.0-6           gtable_0.2.0           git2r_0.21.0          
[37] scales_0.5.0           KernSmooth_2.23-15     stringi_1.1.6         
[40] impute_1.52.0          reshape2_1.4.3         bindrcpp_0.2          
[43] RColorBrewer_1.1-2     rjson_0.2.15           tools_3.4.2           
[46] BSgenome_1.46.0        glue_1.2.0             seqPattern_1.10.0     
[49] hms_0.4.1              plotrix_3.7            yaml_2.1.16           
[52] colorspace_1.3-2       knitr_1.18             bindr_0.1             </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
