---
title: "Create Gviz genome track plots"
author: "Briana Mittleman"
date: 2017-12-13
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->



The goal of this analysis is to create nice plots showing that we are getting as much information as the 1 lane from the Mayer sample. I will do this with our merged data vs. their 1 lane.  

Genes from IGV that I want to use: 

* HERPUD1 chr16:56,964,002-56,979,793  

* ACTB chr7:5,564,779-5,572,232  

* CCNB2 chr15:59,396,707-59,401,006  


* chr11:234,336-239,997 

* KIAA0100 chr17:26,968,078-26,974,887  

* HECTD1 chr14:31,672,040-31,681,043  


* STAG1 chr3:136,469,421-136,472,771  


Load Packages:  
```{r}
library(Gviz)
library(GenomicRanges)
library(biomaRt)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library("IRanges")
library("dplyr")
library("data.table")
library("GenomicAlignments")
```


Upload data:  
```{r}

chr= "chr7"
gen= "hg19"

options(ucscChromosomeNames=FALSE)
merged_data_7= DataTrack(range = "../data/bam_files_chr/merged_Net1_chr.bam", genome = gen, type = "h", name = "Merged", window = -1, chromosome = "chr7")
mayer_data_7= DataTrack(range = "../data/bam_files_chr/mayer_chr.bam", genome = gen, type = "h", name = "Mayer", window = -1, chromosome = "chr7")



refGenes <- UcscTrack(genome= gen, chromosome=chr, 
     track="RefSeq Genes", from = 5555158, to = 5581854, 
     trackType="GeneRegionTrack", rstarts="exonStarts", 
     rends="exonEnds", gene="name", symbol="name2", 
     transcript="name", strand="strand", fill="#800000", name="RefSeq Genes", showId=TRUE)
gtrack = GenomeAxisTrack()


itrack= IdeogramTrack(genome = gen, chromosome = chr)


#plots
plotTracks(list(itrack,gtrack, merged_data_7, mayer_data_7, refGenes), from = 5555158, to = 5581854,  background.title="darkblue",  background.panel = "#FFFEDB")

```

```{r}
merged_anno_7= AnnotationTrack(range = "../data/bam_files_chr/merged_Net1_chr.bam", genome = gen, name = "Merged", window = -1, chromosome = chr)
mayer_anno_7= AnnotationTrack(range = "../data/bam_files_chr/mayer_chr.bam", genome = gen, name = "Mayer", window = -1, chromosome = chr)
plotTracks(list(merged_data_7, merged_anno_7), from = 5564779, to = 5572232)
plotTracks(list(mayer_data_7,mayer_anno_7), from = 5564779, to = 5572232)
```

Read in files with readGAlignments  
```{r, eval=FALSE}
reads <- readGAlignments(file="../data/merged_Net1.bam", index="../data/merged_Net1.bam.bai")
reads.GR <-  granges(reads)
reads.GR <- renameSeqlevels(reads.GR, paste("chr",seqlevels(reads.GR), sep= ""))
```


####Fix in terminal  
Add the chr tag to the bam files.  

```{bash, eval=FALSE}
samtools view -h SRR1575922-sort.bam | awk 'BEGIN{FS=OFS="\t"} (/^@/ && !/@SQ/){print $0} $2~/^SN:[1-9]|^SN:X|^SN:Y|^SN:MT/{print $0}  $3~/^[1-9]|X|Y|MT/{$3="chr"$3; print $0} ' | sed 's/SN:/SN:chr/g' | sed 's/chrMT/chrM/g' | samtools view -bS - > mayer_chr.bam

samtools view -h merged_Net1.bam | awk 'BEGIN{FS=OFS="\t"} (/^@/ && !/@SQ/){print $0} $2~/^SN:[1-9]|^SN:X|^SN:Y|^SN:MT/{print $0}  $3~/^[1-9]|X|Y|MT/{$3="chr"$3; print $0} ' | sed 's/SN:/SN:chr/g' | sed 's/chrMT/chrM/g' | samtools view -bS - > merged_Net1_chr.bam
```

index the bam files:  

```{bash, eval=FALSE}
samtools index mayer_chr.bam
samtools index merged_Net1_chr.bam
```


## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
