---
title: "Home"
output:
  html_document:
    toc: false
---

This is the workflowr site for my Net-seq project. I will have all of the data, code, and analysis here.  


Set-up for mappping my libraries:  

* [Setting up conda environment](conda.environment.html)  

* [Setting up config and snake files](config.snake.setup.html)  


Initial data exploration:  

* [Initial data exploration for NetSeq1 libary](initial.data.exploration.html)  

* [Explore read lengths that are mapping](map_stats_from_bam.html)  

* [Explore the UMI usage in the NetSeq1 library](explore_umi_usage.html)  

* [Use the genomation package to vizualize genome features and see overlap with Net-seq1](visualize_genomefeatures.html)  

* [Exploration of read counts per gene](reads_in_genes.html)  

* [Reformat gene counts for ggplot2 data analysis](data_for_ggplot.html)  

* [Explore read counts per genome bin](bin_windows.html)  

* [Create IGV browser style plots](gviz_plots.html)


Initial analysis for Net-3:  

* [Net-3 Initial exploration](Net_3_explore.html)  


Net-pilot analysis: 

* [Update snakefile for pilot analysis](update_snakefile.html)  

* [Create blacklist to filter reads](create_blacklist.html)  

* [Recreate figures from Mayer](recreate_mayer_figs.html)

* [Strand specific analysis](strand_spec.html)

* [Connect eQTLs to strand](eqtl_bystrand.html)


Cell Growth analysis: 

* [Initial plots on growth](LCL_growth.html)

Exploration of the 3` seq and ApA sites:  

* [Look for read enrichment at ApA sites](three_prime_UTR.html) 

* [UTR coverage analysis](UTR_coverage.html)

* [Look at enrichment using Deep tools](use_deeptools.html)


