---
title: "Initial Data Exploration Netseq1"
author: "Brina Mittleman"
date: 2017-11-06
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

I will use this analysis to look at inital data QC and points of interests.  

First I looked at the number of reads that mapp to the genome before and after deduplication UMI steps.  

`samtools view -c -F 4`  

For flag info: https://broadinstitute.github.io/picard/explain-flags.html  


Mayer data:
fastq: 137281933  
sorted: 120124203
dedup: 2262387  
dedup/sorted: 0.01883373  

```{r}
library= c( "18486-dep", "18486-tot", "18508-dep", "18508-nondep", "19238-dep", "mayer")

fastq= c( 45803834, 4006, 70776230, 77223987, 113160855, 13728193)  

sorted= c(17336796, 1258, 43247747, 50189574, 40420633, 17157730 )

dedup= c(1533069, 1105, 1776330,1919904,
1870359,2262387)

perc= dedup/sorted

reads_mapped_dedup= data.frame(rbind(library, fastq, sorted, dedup, perc))

reads_mapped_dedup

total_reads= sum(fastq)

sorted/fastq
```



Undetermined is nothing: it corresponds to random reads  

From meeting:  

* Allign with star and bwa to compare  

* compare to http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr7%3A5568588%2D5568715&hgsid=642260271_FLEwDANY0lSWCFhW4QjbmbASDDnB  

###Explore UMIs

'samtools view YG-SP-NET1-18486-total-2017-10-13_S5_R1_001-sort.bam | tr "_" "\t" | cut -f 2 | sort | uniq -c > UMI_18486_total_stats.txt '  

'samtools view YG-SP-NET1-18486-total-2017-10-13_S5_R1_001-sort.bam | tr "_" "\t" | cut -f 2 > UMI_18486_total.txt'  

Run this for the files and put in output- then i can make plots based on this.  




## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
