---
title: "Initial Data Exploration Netseq1"
author: "Brina Mittleman"
date: 2017-11-06
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

I will use this analysis to look at inital data QC and points of interests.  

First I looked at the number of reads that mapp to the genome before and after deduplication UMI steps.  

`samtools view -c -f 4`


Mayer data:
fastq: 137281933  
sorted: 120124203
dedup: 2262387  
dedup/sorted: 0.01883373  

```{r}
library= c("und", "18486-dep", "18486-tot", "18508-dep", "18508-nondep", "unk1", "mayer")

fastq= c(3102688, 45803834, 4006, 70776230, 77223987, 113160855, 137281933    )  

sorted= c(1341658 ,28467038, 2748, 27528483, 27034413, 72740222,120124203 )

dedup= c(320894, 1533069, 1105, 1776330,1919904,
1870359,2262387    )

perc= dedup/sorted

reads_mapped_dedup= rbind(library, fastq, sorted, dedup,perc)

reads_mapped_dedup

```


## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
