---
title: "Coverage in gene regions"
author: "Briana Mittleman"
date: 2017-11-20
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

##Initial exploration for data format  


The goal of this analysis is too look at read coverage in genes/exons/and in genes + promoter regions for my Net-seq data and the Mayer data. This will help me understand if the problem with this data is coverage. 

I have downloaded 3 bed files from the UCSC table browser:  

* Ref seq genes  

* Ref seq genes + 250 upstream  

* Ref seq exons  


I will use the bedtools coverage function. Documentation is below:  

http://bedtools.readthedocs.io/en/latest/content/tools/coverage.html  

* a bed file for the genes/exons  
* bam files for my data and the mayer data  

```{bash, eval=FALSE}

bedtools coverage -a <FILE> \
                             -b <FILE1, FILE2, ..., FILEN>
```


Write a bash script for genes overlap:  genome_overlap.sh
```{bash, eval=FALSE}
#!/bin/bash

#SBATCH --job-name=bedtools_coverage
#SBATCH --time=8:00:00
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --tasks-per-node=4 

#$1 reference to use 


bedtools coverage -a $1 -b /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18486-dep-2017-10-13_S4_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18508-dep-2017-10-13_S2_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-18508-nondep-2017-10-13_S3_R1_001-sort.bam /project2/gilad/briana/Net-seq/Net-seq1/data/sort/YG-SP-NET1-Unk1_S6_R1_001-sort.bam /project2/gilad/briana/Net-seq/data/sort/SRR1575922-sort.bam 

#> to a txt file  


bedtools coverage -a ref_seq_gene_hg19_ -b /project2/gilad/briana/Net-seq/Net-seq1/data/net1_18486_dep_dedup_chr.bed -hist  > gene_coverage_18486_dedup_hist.txt

```


Try with only 1 file first.  

Try with /project2/gilad/briana/Net-seq/Net-seq1/data/net1_18486_dep_dedup_chr.bed  

This works but it is for the deduplicated file. I am going to do the sorted files first. I need to convert the sorted bam files to bed files. All of the sorted bed files with the chr label are in data/bed  

Should probably do this for the deduplicated files as well.  

```{bash, eval=FALSE}
bedtools bamtobed [OPTIONS] -i <BAM>
awk '$0="chr"$0' file > new_file 
```



Only want first 6 columns of the ref file  

```{bash, eval=FALSE}


cat ref_seq_gene_hg19  | cut  -f 1-6 > ref_seq_gene_hg19_small_cut

```


Still to big. Run on one file.  

```{bash, eval=FALSE}
bedtools coverage -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed/net1_18486_dep_chr.bed  > gene_coverage_18486_hist.txt
```
Run this to make:  

* gene_coverage_18508_dep_hist.txt  

* gene_coverage_18508_dep_hist.txt  

* gene_coverage_18508_nondep_hist.txt   

* gene_coverage_19238_dep_hist.txt  

* gene_coverage_mayer_SRR1575922_hist.txt  



Sort the bed files then rerun the coverage with the counts option to get one read count per gene. I have the script to sort the script files in each data/bed folder.  ls


```{bash, eval=FALSE}
bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18486_dep_chr_sort.bed  > gene_cov_count/gene_coverage_18486_count.txt


bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18508_dep_chr_sort.bed  > gene_cov_count/gene_coverage_18508_dep_count.txt

bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_18508_nondep_chr_sort.bed   > gene_cov_count/gene_coverage_18508_nondep_count.txt

bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/Net-seq1/data/bed_sort/net1_19238_dep_chr_sort.bed   > gene_cov_count/gene_coverage_19238_dep_count.txt


bedtools coverage -counts -a ref_seq_gene_hg19_small_cut  -b /project2/gilad/briana/Net-seq/data/bed_sort/mayer_SRR1575922_chr_sort.bed   > gene_cov_count/gene_coverage_mayer_SRR1575922_count.txt

```
##Data input (non-depuplicated)

###Gene Coverage 

Install packages:  
```{r}
library(vioplot)
library(dplyr)
library(ggplot2)
```



```{r}
gene_coverage_18486_count= read.csv("../data/gene_cov_count/gene_coverage_18486_count.txt", header=FALSE, sep="\t")

gene_coverage_18508_dep_count= read.csv("../data/gene_cov_count/gene_coverage_18508_dep_count.txt", header=FALSE, sep="\t")

gene_coverage_18508_nondep_count= read.csv("../data/gene_cov_count/gene_coverage_18508_nondep_count.txt", header=FALSE, sep="\t")

gene_coverage_19238_dep_count= read.csv("../data/gene_cov_count/gene_coverage_19238_dep_count.txt", header=FALSE, sep="\t")

gene_coverage_mayer_dep_count = read.csv("../data/gene_cov_count/gene_coverage_mayer_SRR1575922_count.txt", header=FALSE, sep="\t")
```


```{r}
colnames(gene_coverage_18486_count) = c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(gene_coverage_18508_dep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(gene_coverage_18508_nondep_count)=c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(gene_coverage_19238_dep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(gene_coverage_mayer_dep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
```


Next I will merge theses files to create 1 file with all of the data.  

```{r}
gene_coverage_all= cbind(gene_coverage_18486_count, gene_coverage_18508_dep_count$counts, gene_coverage_18508_nondep_count$counts, gene_coverage_19238_dep_count$counts, gene_coverage_mayer_dep_count$counts)
colnames(gene_coverage_all)= c("chr", "start", "end", "name", "score", "strand", "c_18486", "c_18508_dep", "c_18508_nondep", "c_19238_dep", "c_mayer")
```


Add a column for length of A:  

```{r}

#add length column  
gene_coverage_all= mutate(gene_coverage_all, length=end-start)
#add columns for standard count 
gene_coverage_all= mutate(gene_coverage_all,st_18486=c_18486/length)
gene_coverage_all= mutate(gene_coverage_all,st_18508_dep= c_18508_dep /length)
gene_coverage_all= mutate(gene_coverage_all,st_18508_nondep=c_18508_nondep/length)
gene_coverage_all= mutate(gene_coverage_all,st_19238=c_19238_dep/length)
gene_coverage_all= mutate(gene_coverage_all,st_mayer=c_mayer/length)
```




```{r}

plot(sort(-log(gene_coverage_all$st_18486), decreasing=TRUE), ylab="-log of gene count standardized by length", xlab="Genes", main="18486")
abline(a=-log(mean(gene_coverage_all$st_18486)),b=0)
plot(sort(-log(gene_coverage_all$st_18508_dep), decreasing=TRUE), ylab="-log of gene count standardized by length", xlab="Genes", main="18505 dep")
abline(a=-log(mean(gene_coverage_all$st_18508_dep)),b=0)
plot(sort(-log(gene_coverage_all$st_18508_nondep), decreasing=TRUE), ylab="-log of gene count standardized by length", xlab="Genes", main="18505 nondep")
abline(a=-log(mean(gene_coverage_all$st_18508_nondep)),b=0)
plot(sort(-log(gene_coverage_all$st_19238), decreasing=TRUE), ylab="-log of gene count standardized by length", xlab="Genes", main="19238")
abline(a=-log(mean(gene_coverage_all$st_19238)),b=0)
plot(sort(-log(gene_coverage_all$st_mayer), decreasing=TRUE), ylab="-log of gene count standardized by length", xlab="Genes", main="Mayer")
abline(a=-log(mean(gene_coverage_all$st_mayer)),b=0)


```

```{r}

#boxplot- violin plot is better but you get infinities  
boxplot(-log(gene_coverage_all$st_18486), -log(gene_coverage_all$st_18508_dep), -log(gene_coverage_all$st_18508_nondep), -log(gene_coverage_all$st_19238), -log(gene_coverage_all$st_mayer), names=c("18486", "18508_dep", "18508_nondep", "19238", "Mayer"), ylab="-log of standard expression")


```





To use ggplot I need to reformat the dataframe to gene by sample.   

```{r}
gene_names= gene_coverage_all$name
standard_counts= gene_coverage_all[, 13:17]
#row.names(standard_counts)=gene_names
```

```{r}


```




###Gene and promoter coverage  


```{r}
prom_coverage_18486_count= read.csv("../data/prom_coverage/prom_coverage_18486_count.txt", header=FALSE, sep="\t")

prom_coverage_18508_dep_count= read.csv("../data/prom_coverage/prom_coverage_18508_dep_count.txt", header=FALSE, sep="\t")

prom_coverage_18508_nondep_count= read.csv("../data/prom_coverage/prom_coverage_18508_nondep_count.txt", header=FALSE, sep="\t")

prom_coverage_19238_dep_count= read.csv("../data/prom_coverage/prom_coverage_19238_dep_count.txt", header=FALSE, sep="\t")

prom_coverage_mayer_dep_count = read.csv("../data/prom_coverage/promoter_coverage_mayer_SRR1575922_count.txt", header=FALSE, sep="\t")
```

```{r}
colnames(prom_coverage_mayer_dep_count)=  c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(prom_coverage_19238_dep_count)=  c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(prom_coverage_18508_nondep_count)=  c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(prom_coverage_18508_dep_count) =  c("chr", "start", "end", "name", "score", "strand", "counts")  
colnames(prom_coverage_18486_count)=  c("chr", "start", "end", "name", "score", "strand", "counts")

```




###Exon coverage  

```{r}
exon_coverage_18486_count= read.csv("../data/exon_cov/exon_coverage_18486_count.txt", header=FALSE, sep="\t")

exon_coverage_18508_dep_count= read.csv("../data/exon_cov/exon_coverage_18508_dep_count.txt", header=FALSE, sep="\t")

exon_coverage_18508_nondep_count= read.csv("../data/exon_cov/exon_coverage_18508_nondep_count.txt", header=FALSE, sep="\t")

exon_coverage_19238_dep_count= read.csv("../data/exon_cov/exon_coverage_19238_dep_count.txt", header=FALSE, sep="\t")

exon_coverage_mayer_dep_count = read.csv("../data/exon_cov/exon_coverage_mayer_SRR1575922_count.txt", header=FALSE, sep="\t")
```

```{r}
colnames(exon_coverage_mayer_dep_count)=c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(exon_coverage_19238_dep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(exon_coverage_18508_nondep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(exon_coverage_18508_dep_count)= c("chr", "start", "end", "name", "score", "strand", "counts")
colnames(exon_coverage_18486_count)= c("chr", "start", "end", "name", "score", "strand", "counts")  
```


###Coverage with histograms  
These are the historgram files I made.  

```{r}
#gene_coverage_18486_dedup= read.csv("../data/gene_coverage_18486_dedup_hist.txt", header=FALSE, sep="\t")


#gene_coverage_18486= read.csv("../data/gene_coverage_18486_hist.txt", head=FALSE, sep="\t")

#gene_coverage_18508_dep= read.csv("../data/gene_coverage_18508_dep_hist.txt", header=FALSE, sep="\t")
#gene_coverage_18508_nondep= read.csv("../data/gene_coverage_18508_nondep_hist.txt", head=FALSE, sep="\t")
#gene_coverage_19238_dep = read.csv("../data/gene_coverage_19238_dep_hist.txt", head=FALSE, sep="\t")
#gene_coverage_mayer_dep = read.csv("../data/gene_coverage_mayer_SRR1575922_hist.txt", head=FALSE, sep="\t")


```
Add column names:  

```{r}
#colnames(gene_coverage_18486_dedup)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")
#colnames(gene_coverage_18486)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")
#colnames(gene_coverage_18508_dep)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")
#colnames(gene_coverage_18508_nondep)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")
#colnames(gene_coverage_19238_dep)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")
#colnames(gene_coverage_mayer_dep)= c("chr", "start", "end", "name", "score", "strand", "overlap", "bases_non_zero", "lengthA", "frac_A_non_zero_hist")

```

Omit NA columns:  

```{r}
# gene_coverage_18486= na.omit(gene_coverage_18486)
# gene_coverage_18486_dedup= na.omit(gene_coverage_18486_dedup)
# gene_coverage_18508_dep= na.omit(gene_coverage_18508_hist_dep)
# gene_coverage_18508_nondep= na.omit(gene_coverage_18508_nondep)
# gene_coverage_19238_dep= na.omit(gene_coverage_19238_dep)
# gene_coverage_mayer_dep= na.omit(gene_coverage_mayer_dep)

```



## Session information

<!-- Insert the session information into the document -->
```{r session-info}

```
