---
title: "3` UTR analysis"
author: "Briana Mittleman"
date: 2018-01-24
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->



###Called ApA analysis  

Alternative anaylsis: Try to look at this similar to how I looked at the TSS enrichment. 

* Step 1: Download drop and dronc seq bam/ index files to my computer in the netseq data file.  

- clusters.hg38  

- Day7_cardiomyocytes_drop_seq.bam  

- Day7_cardiomyocytes_drop_seq.bam.bai  

- three_prime_utr.bed  


* Step 2: Pull in packages and data for analysis:  


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("IRanges")
library("GenomicRanges")
library("dplyr")
library("genomation")
library("data.table")
library(magrittr)
source("https://bioconductor.org/biocLite.R")
#biocLite("GenomicAlignments")
library("GenomicAlignments")
```

```{r}
#get reads
reads <- readGAlignments(file = "../data/Day7_cardiomyocytes_drop_seq.bam", index="../data/Day7_cardiomyocytes_drop_seq.bam.bai")
reads.GR <-  granges(reads)

UTR=readGeneric("../data/three_prime_utr.bed")
pAsite= readGeneric("../data/clusters.hg38.bed")

```




```{r}
#resize so I am looking 10,000 up and downstream of the center of the UTR
UTR %<>% resize(., width=10000, fix="center")
(UTR_width= summary(width(UTR)))
sm <- ScoreMatrixBin(target = reads.GR,  windows = UTR, bin.num = 100, bin.op = "mean")  
plotMeta(sm)


```


Do this against the pAsites: 

```{r}
(pAs_width= summary(width(pAsite)))
#look 200 up and down stream of each 
pAsite %<>% resize(., width=500, fix="center")
(pAs_width2= summary(width(pAsite)))
sm_pA <- ScoreMatrixBin(target = reads.GR,  windows = pAsite, bin.num = 500, bin.op = "mean",strand.aware=TRUE)
plotMeta(sm_pA, xcoords = c(-250,250), main="pA enrichment Day7 Drop")

```

Try to make this a heat map:  

```{r}
sm_pA_heat <- ScoreMatrix(target = reads.GR,  windows = pAsite,strand.aware=TRUE)
heatMatrix(sm_pA_heat)

```







Try with the Dronc seq data:  
```{r}
dronc_reads <- readGAlignments(file = "../data/Day7_cardiomyocytes_droNC_seq.bam", index="../data/Day7_cardiomyocytes_droNC_seq.bam.bai")
dronc_reads.GR <-  granges(dronc_reads)


dronc_sm_pA <- ScoreMatrixBin(target =dronc_reads.GR,  windows = pAsite, bin.num = 500, bin.op = "mean", strand.aware=TRUE)
plotMeta(dronc_sm_pA, xcoords = c(-250,250),  main="pA enrichment Day7 Dronc")
```




Compare this result to the 3` seq data :  


* LCL  


```{r}
LCL_reads <- readGAlignments(file = "../data/blcl.hg38.sorted.bam", index="../data/blcl.hg38.sorted.bam.bai")
LCL_reads.GR <-  granges(LCL_reads)

sm_LCL_pA <- ScoreMatrixBin(target = LCL_reads.GR,  windows = pAsite, bin.num = 500, bin.op = "mean", strand.aware=TRUE)
plotMeta(sm_LCL_pA, xcoords = c(-250,250),  main="pA enrichment LCL")
```

```{r}
sm_LCL_heat <- ScoreMatrix(target = LCL_reads.GR,  windows = pAsite, strand.aware=TRUE)
heatMatrix(sm_LCL_heat)
```



* hES  

```{r}
hES_reads <- readGAlignments(file = "../data/hES.hg38.sorted.bam", index="../data/hES.hg38.sorted.bam.bai") 
hES_reads.GR <-  granges(hES_reads)

sm_hES_pA <- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite, bin.num = 500, bin.op = "mean", strand.aware=TRUE) 
plotMeta(sm_hES_pA, xcoords = c(-250,250),  main="pA enrichment hES") 
``` 

I now want to try to look at the 3` most pAs. Start by overlapping the PaS sites with the UTRs then annotate the file with the UTR name.  

bedtools intersect- I want only the As (pAs) that are in B (UTR)

* bedtools intersect -a clusters.hg38.bed -b three_prime_utr.bed > clusters.hg38.3utr.bed  

Alternative methed: subset the pAS in the 3' UTRs then seperate the files by strandedness.  

```{r}
pAsite_pos= readGeneric("../data/clusters.hg38.3utr.pos.bed")
pAsite_pos %<>% resize(., width=1000, fix="center")
(pAs_pos_width= summary(width(pAsite_pos)))
pAsite_neg= readGeneric("../data/clusters.hg38.3utr.neg.bed")
pAsite_neg %<>% resize(., width=1000, fix="center")




```



```{r}
#drop and pos strand
sm_pA_pos <- ScoreMatrixBin(target = reads.GR,  windows = pAsite_pos, bin.num = 1000, bin.op = "mean")
plotMeta(sm_pA_pos, xcoords = c(-500,500), main="pA enrichment Day7 Drop pos strand")
```
negative Graph:  

```{r}
#drop and neg strand
sm_pA_neg <- ScoreMatrixBin(target = reads.GR,  windows = pAsite_neg, bin.num = 1000, bin.op = "mean")
plotMeta(sm_pA_neg, xcoords = c(-500,500), main="pA enrichment Day7 Drop neg strand")
```

hES with pos

```{r}
sm_hES_pA_pos <- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite_pos, bin.num = 1000, bin.op = "mean") 
plotMeta(sm_hES_pA_pos, xcoords = c(-500,500),  main="pA enrichment hES pos") 
```

hES neg  

```{r}
sm_hES_pA_neg <- ScoreMatrixBin(target = hES_reads.GR,  windows = pAsite_neg, bin.num = 1000, bin.op = "mean") 
plotMeta(sm_hES_pA_neg, xcoords = c(-500,500),  main="pA enrichment hES pos") 
```

###Heat maps with EnrichedHeatmap  

```{r}
library(EnrichedHeatmap)
```
Drop seq day7 cardiomyocytes GRanges object:  

```{r}
reads.GR[1:5]
```

pAs:  

```{r}
pAsite[1:5]
```

```{r}

#mat_drop7_pAs = normalizeToMatrix(UTR , pAsite, value_column = "ranges", extend = 1000, mean_mode = "w0", w = 100)

```




```{r}
# library(biomaRt)
# mart = useMart(biomart = "ENSEMBL_MART_ENSEMBL",
#                dataset = "hsapiens_gene_ensembl")
# filterlist <- c(1:22, 'X', 'Y')
# ds = useDataset('hsapiens_gene_ensembl', mart = mart)
# 
# egs = getBM(attributes = c('ensembl_gene_id', 'external_gene_name', 'chromosome_name', 'start_position', 'end_position', 'strand'), 
#             filters = 'chromosome_name',
#             values = filterlist,
#             mart = ds)
# reads.GR_chr <-  granges(reads, seqnames = Rle(paste0('chr', egs$chromosome_name )))
# pAsite_chr= readGeneric("../data/clusters.hg38.bed", seqnames = Rle(paste0('chr', egs$chromosome_name )))
```


## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
