---
title: "Creat heatplots with deeptools"
author: "First Last"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->


###Deeptools  

Deeptools is a way to make plots similar to genomation but they have enrichment plots and heatmaps together. Also this is in bash so I do not need to get the bam files into R.  

####Install  
I added the deeptools package to my net-seq conda environment.  

####Running:

Step 1: Create bigwig coverage files with bamcoverage 

* bamCoverage -b reads.bam -o coverage.bw

Step 2: computeMatrix  

* I will need my normalized bigwig reads and the bed interval file (in my case PAS clusters)

* ex: computeMatrix scale-regions -S <biwig file(s)> -R <bed file> -b 1000 -a 1000 -out <compressed matrix name.gz>

* --skipZeros  (option- not included in first try)

Step 3: Plot heatmap  

* required –matrixFile, -m (from the compute matrix), -out (file name to save image.png)

* –sortRegions descending  

* –plotTitle, -T 


I will run this first on the hES 3' seq becasue we expect heavy enrichment.  

```{bash, eval=FALSE}
#!/bin/bash


#SBATCH --job-name=deeptools_hES
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3  

source activate net-seq  

bamCoverage -b /project2/gilad/briana/Lianoglou.data/hES.hg38.sorted.bam -o hES.cov.bw  

computeMatrix reference-point -S hES.cov.bw -R /project2/gilad/briana/apa_sites/clusters.hg38.bed -b 1000 -a 1000 -out hES.hg38.gz

plotHeatmap --sortRegions descend -m hES.hg38.gz  -out hES.hg38.apa.png 

```

This plot shows enrichement of the 3' seq on called PAS sites. The axis are wrong.  

![hES 3' coverage at APA sites](../data/hES.hg38.apa.png)

I will make this plot excluding exons. I will use bedtools intersect to get APA sites not in exons.  

I want A interects B -v. This will give me regions in A not in B. A is the cluster file. /project2/gilad/briana/apa_sites/clusters.hg38.bed and B is /project2/gilad/briana/apa_sites/deeptools/exons_hg38.bed 


This bash file is /project2/gilad/briana/apa_sites/deeptools/heatmatrix_hES_apa_excludeExons.sh 
```{bash, eval=FALSE}
#!/bin/bash


#SBATCH --job-name=deeptools_hES_noexon
#SBATCH --time=8:00:00
#SBATCH --partition=gilad
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3  

source activate net-seq  

bedtools intersect -v -a /project2/gilad/briana/apa_sites/clusters.hg38.bed -b /project2/gilad/yangili/hg38_exons.bed > ../clusters.hg38.noExons.bed 


#bamCoverage -b /project2/gilad/briana/Lianoglou.data/hES.hg38.sorted.bam -o hES.cov.bw
# dont need to redo this. The file exsists

computeMatrix reference-point -S hES.cov.bw -R /project2/gilad/briana/apa_sites/clusters.hg38.noExons.bed -b 1000 -a 1000 -out hES.hg38.noExons.gz

plotHeatmap --sortRegions descend -T "hES 3` APA enrichment exclude exons" -m hES.hg38.noExons.gz  -out hES.hg38.apa.noExons.png 

```
![hES 3' coverage at APA sites not in exons](../data/hES.hg38.apa.noExons.png)

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
