---
title: "Use the genomation package to vizualize genome features and see overlap with Net-seq1"
author: "Briana Mittleman"
date: 2017-11-09
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

I will look at genome feature overlap with the Netseq1 libary. I will start with one sample then move to more once the code is working. I will look specifically at overlap with CTCF sites. 


Load libraries:  
```{r}
library(genomation)
library(genomationData)
```


###Load gata: 

```{r}
CTCF_gen= readGeneric(file = "../data/hg19.GM72.CTCF", header=FALSE, keep.all.metadata = TRUE)
#bed file for one of my libraries  
# to get from midway
#scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/Net-seq/Net-seq1/data/cov/YG-SP-NET1-18486-dep-2017-10-13_S4_R1_001-sort.dedup.cov.bed .

dedup.18486.dep= readGeneric(file= "../data/net1_18486_dep_dedup_chr.bed", header = FALSE, keep.all.metadata = TRUE)



```


###Data extaction:  

Try to get the density of net-seq reads around CTCF sites  

fix formating of file with 

```{bash, eval=FALSE }
#awk '$0="chr"$0' file  > newfile
```

Need to do this with the reads not the coverage file. I used the samtools bamtobed function on one of the dedup files and copied this to the data file here.  


```{r}
peak.annot = annotateWithFeature(CTCF_gen, dedup.18486.dep, intersect.chr = TRUE)
peak.annot


```
I need to compare this to another feature. I will look at the genes file.  

```{r}
genes= readGeneric(file= "../data/hg19.ref.genes.bed", header = FALSE, keep.all.metadata = TRUE)
head(genes)

peak.annot_genes = annotateWithFeature(genes, dedup.18486.dep, intersect.chr = TRUE)
peak.annot_genes
```
The genes file has a lot of random chr tags at the end.  


```{r}

sm = ScoreMatrixBin(target = dedup.18486.dep, windows = CTCF_gen, bin.num = 50)
heat.matrix= heatMatrix(sm, xcoords = c(-1000, 1000))
heat.matrix

plotMeta(sm, xcoords = c(-1000, 1000))
```


###Metaplot for TSS  

```{r}
TSS= readGeneric(file = "../data/SwitchGear_TSS.bed", header=FALSE, keep.all.metadata = TRUE)


dedup.18486.dep= readGeneric(file= "../data/net1_18486_dep_dedup_chr.bed", header = FALSE, keep.all.metadata = TRUE)


annotateWithFeature(TSS, dedup.18486.dep, intersect.chr = TRUE)


#sm2 = ScoreMatrixBin(target = dedup.18486.dep, windows = TSS)
#plotMeta(sm2, xcoords = c(-1000, 1000))



```
Problem: This TSS file just has the 1bp site. I need to make a file that is 200 up and downstream of this site. Maybe first try the promoter file that was 250bp upstream of the gene.  


```{r}
ref_250up= readGeneric(file = "../data/refseq_250up.bed", header=FALSE, keep.all.metadata = TRUE)


dedup.18486.dep= readGeneric(file= "../data/net1_18486_dep_dedup_chr.bed", header = FALSE, keep.all.metadata = TRUE)


sm2 = ScoreMatrixBin(target = dedup.18486.dep, windows = ref_250up)
plotMeta(sm2, xcoords = c(-1000, 1000), main="Windows are 250bp upstream")


```



## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
